# Biological theme comparison {#clusterprofiler-comparecluster}



```{r include=FALSE}
library(knitr)
opts_chunk$set(message=FALSE, warning=FALSE, eval=TRUE, echo=TRUE, cache=TRUE)
library(clusterProfiler)
```

The `r Biocpkg("clusterProfiler")` package was developed for biological theme comparison [@yu2012], and it provides a function, `compareCluster`, to automatically calculate enriched functional categories of each gene clusters. 

## Comparing multiple gene lists

The `compareCluster()` function applies selected function (via the `fun` parameter) to perform enrichment analysis for each gene list. All the results for the input gene lists will be aggregated into a single object. 

```{r}
data(gcSample)
str(gcSample)
```

The input for `geneCluster` parameter should be a named list of gene IDs. 

```{r}
ck <- compareCluster(geneCluster = gcSample, fun = enrichKEGG)
ck <- setReadable(ck, OrgDb = org.Hs.eg.db, keyType="ENTREZID")
head(ck)
```


## Formula interface of compareCluster

As an alternaitve to using named list, the `compareCluster()` function also supports passing a formula of type $Entrez \sim group$ or $Entrez \sim group + othergroup$.


```{r}
mydf <- data.frame(Entrez=names(geneList), FC=geneList)
mydf <- mydf[abs(mydf$FC) > 1,]
mydf$group <- "upregulated"
mydf$group[mydf$FC < 0] <- "downregulated"
mydf$othergroup <- "A"
mydf$othergroup[abs(mydf$FC) > 2] <- "B"

formula_res <- compareCluster(Entrez~group+othergroup, data=mydf, fun="enrichKEGG")

head(formula_res)
```

## Visualization of profile comparison

We can visualize the result using the `dotplot()` method.



(ref:compareClusterGeneListscap) Comparing enrichment results of multiple gene lists.

(ref:compareClusterGeneListcap) **Comparing enrichment results of multiple gene lists.** (A) Using a named list of gene clusters, the results were displayed as multiple columns with each one represents an enrichment result of a gene cluster. (B) Using formula interface, the columns represent gene clusters defined by the formula.  

```r
dotplot(ck)
dotplot(formula_res)
```


```{r comparecluster-genelists, fig.height=8, fig.width=20, fig.cap="(ref:compareClusterGeneListcap)", fig.scap="(ref:compareClusterGeneListscap)", echo=FALSE}
library(stringr)
library(ggplot2)
p1 <- dotplot(ck) + scale_y_discrete(labels=function(x) str_wrap(x, width=50)) + scale_size(range = c(3, 10))
p2 <- dotplot(formula_res) + scale_y_discrete(labels=function(x) str_wrap(x, width=30)) + scale_size(range = c(5, 15))
cowplot::plot_grid(p1, p2, ncol=2, labels=c("A", "B"))
```

The fomula interface allows more complicated gene cluster definition. In Figure \@ref(fig:comparecluster-genelists)B, the gene clusters were defined by two variables (i.e. `group` that divides genes into `upregulated` and `downregulated` and `othergroup` that divides the genes into two categories of `A` and `B`.). The `dotplot()` function allows us to use one variable to divide the result into different facet and plot the result with other variables in each facet panel (Figure \@ref(fig:comparecluster-formula)).


(ref:compareClusterFormulascap) Comparing enrichment results of multiple gene lists defined by multiple variable.

(ref:compareClusterFormulacap) **Comparing enrichment results of multiple gene lists defined by multiple variable.**  

```{r comparecluster-formula, fig.height=6, fig.width=12,fig.cap="(ref:compareClusterFormulacap)", fig.scap="(ref:compareClusterFormulascap)"}
dotplot(formula_res, x=~group) + facet_grid(~othergroup)
```


By default, only top 5 (most significant) categories of each cluster
was plotted. User can changes the parameter `showCategory` to
specify how many categories of each cluster to be plotted, and if
`showCategory` was set to `NULL`, the whole result will
be plotted.

The `dotplot()` function accepts a parameter `by` for setting the scale of dot sizes. The default parameter `by` is setting to `geneRatio`, which corresponding to the `GeneRatio` column of the output. If it was setting to `count`, the comparison will be based on gene counts, while if setting to `rowPercentage`, the dot sizes will be normalized by `count/(sum of each row)`

To provide the full information, we also provide number of identified genes in each category (numbers in parentheses) when `by` is setting to `rowPercentage` and number of gene clusters in each cluster label (numbers in parentheses) when `by` is setting to `geneRatio`, as shown in Figure \@ref(fig:comparecluster-genelists). 

The p-values indicate that which categories are more likely to have biological meanings. The dots in the plot are color-coded based on their corresponding adjusted p-values. Color gradient ranging from red to blue correspond to the order of increasing adjusted p-values. That is, red indicate low p-values (high enrichment), and blue indicate high p-values (low enrichment). Adjusted p-values were filtered out by the threshold giving by
parameter `pvalueCutoff`, and FDR can be estimated by `qvalue`.

In [@yu2012], we analyzed the publicly available expression dataset of breast tumour tissues from 200 patients (GSE11121, Gene Expression Omnibus) [@schmidt2008]. We identified 8 gene clusters from differentially expressed genes, and using the `compareCluster()` function to compare these gene clusters by their enriched biological process.

The comparison function was designed as a framework for comparing gene clusters of any kind of ontology associations, not only `=[groupGO](#go-classification), [enrichGO](#clusterprofiler-go-ora), [enrichKEGG](#clusterprofiler-kegg-pathway-ora), [enrichMKEGG](#clusterprofiler-kegg-module-ora), [enrichWP](#clusterprofiler-wikipathway-ora) and [enricher](#universal-api) that were provided in this package, but also other biological and biomedical ontologies, including but not limited to [enrichPathway](#reactomepa-ora), [enrichDO](#dose-do-ora), [enrichNCG](#dose-ncg-ora), [enrichDGN](#dose-dgn-ora) and [enrichMeSH](#meshes-ora). 

