<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>References | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
<meta name="author" content="Guangchuang Yu">
<!-- CSS --><!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.6/header-attrs.js"></script><script src="libs/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.5.2/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.5.2/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.1.9000/tabs.js"></script><script src="libs/bs3compat-0.2.1.9000/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">üìñ Introduction</a></li>
<li class="book-part">Part I: Semantic Similarity Analysis</li>
<li><a class="" href="semantic-similarity-overview.html"><span class="header-section-number">1</span> Overview of semantic similarity analysis</a></li>
<li><a class="" href="GOSemSim.html"><span class="header-section-number">2</span> GO semantic similarity analysis</a></li>
<li><a class="" href="DOSE-semantic-similarity.html"><span class="header-section-number">3</span> DO semantic similarity analysis</a></li>
<li><a class="" href="meshes-semantic-similarity.html"><span class="header-section-number">4</span> MeSH semantic similarity analysis</a></li>
<li class="book-part">Part II: Enrichment analysis</li>
<li><a class="" href="enrichment-overview.html"><span class="header-section-number">5</span> Overview of enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-go.html"><span class="header-section-number">6</span> GO enrichment analysis</a></li>
<li><a class="" href="clusterprofiler-kegg.html"><span class="header-section-number">7</span> KEGG enrichment analysis</a></li>
<li><a class="" href="wikipathways-analysis.html"><span class="header-section-number">8</span> WikiPathways analysis</a></li>
<li><a class="" href="reactomepa.html"><span class="header-section-number">9</span> Reactome enrichment analysis</a></li>
<li><a class="" href="dose-enrichment.html"><span class="header-section-number">10</span> Disease enrichment analysis</a></li>
<li><a class="" href="meshes-enrichment.html"><span class="header-section-number">11</span> MeSH enrichment analysis</a></li>
<li><a class="" href="universal-api.html"><span class="header-section-number">12</span> Universal enrichment analysis</a></li>
<li><a class="" href="chipseeker-enrichment.html"><span class="header-section-number">13</span> Functional enrichment analysis of genomic coordinations</a></li>
<li><a class="" href="clusterprofiler-comparecluster.html"><span class="header-section-number">14</span> Biological theme comparison</a></li>
<li><a class="" href="enrichplot.html"><span class="header-section-number">15</span> Visualization of functional enrichment result</a></li>
<li class="book-part">Part III: Miscellaneous topics</li>
<li><a class="" href="clusterProfiler-dplyr.html"><span class="header-section-number">16</span> dplyr verbs for manipulating enrichment result</a></li>
<li><a class="" href="useful-utilities.html"><span class="header-section-number">17</span> Useful utilities</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="active" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="references" class="section level1 unnumbered">
<h1>References<a class="anchor" aria-label="anchor" href="#references"><i class="fas fa-link"></i></a>
</h1>

<div id="refs" class="references hanging-indent">
<div id="ref-omer_ncg">
<p>A., Omer, Giovanni M. D., Thanos P. M., and Francesca D. C. 2016. ‚ÄúNCG 5.0: Updates of a Manually Curated Repository of Cancer Genes and Associated Properties from Cancer Mutational Screenings.‚Äù <em>Nucleic Acids Research</em> 44 (D1): D992‚ÄìD999. <a href="https://doi.org/10.1093/nar/gkv1123">https://doi.org/10.1093/nar/gkv1123</a>.</p>
</div>
<div id="ref-boyle2004">
<p>Boyle, Elizabeth I, Shuai Weng, Jeremy Gollub, Heng Jin, David Botstein, J Michael Cherry, and Gavin Sherlock. 2004. ‚ÄúGO::TermFinder‚Äìopen Source Software for Accessing Gene Ontology Information and Finding Significantly Enriched Gene Ontology Terms Associated with a List of Genes.‚Äù <em>Bioinformatics (Oxford, England)</em> 20 (18): 3710‚Äì5. <a href="https://doi.org/10.1093/bioinformatics/bth456">https://doi.org/10.1093/bioinformatics/bth456</a>.</p>
</div>
<div id="ref-Du15062009">
<p>Du, Pan, Gang Feng, Jared Flatow, Jie Song, Michelle Holko, Warren A. Kibbe, and Simon M. Lin. 2009. ‚ÄúFrom Disease Ontology to Disease-Ontology Lite: Statistical Methods to Adapt a General-Purpose Ontology for the Test of Gene-Ontology Associations.‚Äù <em>Bioinformatics</em> 25 (12): i63‚Äìi68. <a href="https://doi.org/10.1093/bioinformatics/btp193">https://doi.org/10.1093/bioinformatics/btp193</a>.</p>
</div>
<div id="ref-janet_disgenet">
<p>Janet, P., N√∫ria Q. R., √Älex B., Jordi D. P., Anna B. M., Martin B., Ferran S., and Laura I. F. 2015. ‚ÄúDisGeNET: A Discovery Platform for the Dynamical Exploration of Human Diseases and Their Genes.‚Äù <em>Database</em> 2015 (March): bav028. <a href="https://doi.org/10.1093/database/bav028">https://doi.org/10.1093/database/bav028</a>.</p>
</div>
<div id="ref-jiang_semantic_1997">
<p>Jiang, Jay J., and David W. Conrath. 1997. ‚ÄúSemantic Similarity Based on Corpus Statistics and Lexical Taxonomy.‚Äù <em>Proceedings of 10th International Conference on Research in Computational Linguistics</em>. <a href="http://www.citebase.org/abstract?id=oai:arXiv.org:cmp-lg/9709008">http://www.citebase.org/abstract?id=oai:arXiv.org:cmp-lg/9709008</a>.</p>
</div>
<div id="ref-korotkevich_fast_2019">
<p>Korotkevich, Gennady, Vladimir Sukhov, and Alexey Sergushichev. 2019. ‚ÄúFast Gene Set Enrichment Analysis.‚Äù <em>bioRxiv</em>, October, 060012. <a href="https://doi.org/10.1101/060012">https://doi.org/10.1101/060012</a>.</p>
</div>
<div id="ref-lin_information-theoretic_1998">
<p>Lin, Dekang. 1998. ‚ÄúAn Information-Theoretic Definition of Similarity.‚Äù <em>In Proceedings of the 15th International Conference on Machine Learning</em>, 296‚Äî304. <a href="https://doi.org/10.1.1.55.1832">https://doi.org/10.1.1.55.1832</a>.</p>
</div>
<div id="ref-luo_pathview">
<p>Luo, Weijun, and Cory Brouwer. 2013. ‚ÄúPathview: An R/Bioconductor Package for Pathway-Based Data Integration and Visualization.‚Äù <em>Bioinformatics</em> 29 (14): 1830‚Äì1. <a href="https://doi.org/10.1093/bioinformatics/btt285">https://doi.org/10.1093/bioinformatics/btt285</a>.</p>
</div>
<div id="ref-philip_semantic_1999">
<p>Philip, Resnik. 1999. ‚ÄúSemantic Similarity in a Taxonomy: An Information-Based Measure and Its Application to Problems of Ambiguity in Natural Language.‚Äù <em>Journal of Artificial Intelligence Research</em> 11: 95‚Äì130.</p>
</div>
<div id="ref-schlicker_new_2006">
<p>Schlicker, Andreas, Francisco S Domingues, J√∂rg Rahnenf√ºhrer, and Thomas Lengauer. 2006. ‚ÄúA New Measure for Functional Similarity of Gene Products Based on Gene Ontology.‚Äù <em>BMC Bioinformatics</em> 7: 302. <a href="https://doi.org/1471-2105-7-302">https://doi.org/1471-2105-7-302</a>.</p>
</div>
<div id="ref-schmidt2008">
<p>Schmidt, Marcus, Daniel B?hm, Christian von T?rne, Eric Steiner, Alexander Puhl, Henryk Pilch, Hans-Anton Lehr, Jan G. Hengstler, Heinz K?lbl, and Mathias Gehrmann. 2008. ‚ÄúThe Humoral Immune System Has a Key Prognostic Impact in Node-Negative Breast Cancer.‚Äù <em>Cancer Research</em> 68 (13): 5405‚Äì13. <a href="https://doi.org/10.1158/0008-5472.CAN-07-5206">https://doi.org/10.1158/0008-5472.CAN-07-5206</a>.</p>
</div>
<div id="ref-schriml_disease_2011">
<p>Schriml, L. M., C. Arze, S. Nadendla, Y.-W. W. Chang, M. Mazaitis, V. Felix, G. Feng, and W. A. Kibbe. 2011. ‚ÄúDisease Ontology: A Backbone for Disease Semantic Integration.‚Äù <em>Nucleic Acids Research</em> 40 (D1): D940‚ÄìD946. <a href="https://doi.org/10.1093/nar/gkr972">https://doi.org/10.1093/nar/gkr972</a>.</p>
</div>
<div id="ref-subramanian_gene_2005">
<p>Subramanian, Aravind, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, et al. 2005. ‚ÄúGene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.‚Äù <em>Proceedings of the National Academy of Sciences of the United States of America</em> 102 (43): 15545‚Äì50. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.</p>
</div>
<div id="ref-wang_new_2007">
<p>Wang, James Z, Zhidian Du, Rapeeporn Payattakool, Philip S Yu, and Chin-Fu Chen. 2007. ‚ÄúA New Method to Measure the Semantic Similarity of Go Terms.‚Äù <em>Bioinformatics (Oxford, England)</em> 23 (May): 1274‚Äì81. <a href="https://doi.org/btm087">https://doi.org/btm087</a>.</p>
</div>
<div id="ref-yu_gosemsim_2020">
<p>Yu, Guangchuang. 2020. ‚ÄúGene Ontology Semantic Similarity Analysis Using GOSemSim.‚Äù <em>Methods in Molecular Biology (Clifton, N.J.)</em> 2117: 207‚Äì15. <a href="https://doi.org/10.1007/978-1-0716-0301-7_11">https://doi.org/10.1007/978-1-0716-0301-7_11</a>.</p>
</div>
<div id="ref-yu_meshes_2018">
<p>‚Äî‚Äî‚Äî. 2018. ‚ÄúUsing Meshes for MeSH Term Enrichment and Semantic Analyses.‚Äù <em>Bioinformatics</em> 34 (21): 3766‚Äì7. <a href="https://doi.org/10.1093/bioinformatics/bty410">https://doi.org/10.1093/bioinformatics/bty410</a>.</p>
</div>
<div id="ref-yu_reactomepa_2016">
<p>Yu, Guangchuang, and Qing-Yu He. 2016. ‚ÄúReactomePA: An R/Bioconductor Package for Reactome Pathway Analysis and Visualization.‚Äù <em>Molecular BioSystems</em> 12 (2): 477‚Äì79. <a href="https://doi.org/10.1039/C5MB00663E">https://doi.org/10.1039/C5MB00663E</a>.</p>
</div>
<div id="ref-yu2010">
<p>Yu, Guangchuang, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu, and Shengqi Wang. 2010. ‚ÄúGOSemSim: An R Package for Measuring Semantic Similarity Among Go Terms and Gene Products.‚Äù <em>Bioinformatics</em> 26: 976‚Äì78. <a href="https://doi.org/10.1093/bioinformatics/btq064">https://doi.org/10.1093/bioinformatics/btq064</a>.</p>
</div>
<div id="ref-yu2012">
<p>Yu, Guangchuang, Li-Gen Wang, Yanyan Han, and Qing-Yu He. 2012. ‚ÄúclusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters.‚Äù <em>OMICS: A Journal of Integrative Biology</em> 16 (5): 284‚Äì87. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>.</p>
</div>
<div id="ref-yu_chipseeker_2015">
<p>Yu, Guangchuang, Li-Gen Wang, and Qing-Yu He. 2015. ‚ÄúChIPseeker: An R/Bioconductor Package for Chip Peak Annotation, Comparison and Visualization.‚Äù <em>Bioinformatics</em> 31 (14): 2382‚Äì3. <a href="https://doi.org/10.1093/bioinformatics/btv145">https://doi.org/10.1093/bioinformatics/btv145</a>.</p>
</div>
<div id="ref-yu_dose_2015">
<p>Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. ‚ÄúDOSE: An R/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.‚Äù <em>Bioinformatics</em> 31 (4): 608‚Äì9. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>.</p>
</div>
</div>
</div>






  <div class="chapter-nav">
<div class="prev"><a href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#references">References</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Biomedical Knowledge Mining using GOSemSim and clusterProfiler</strong>" was written by Guangchuang Yu. It was last built on 2021-02-06.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
