<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 KEGG enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
  <meta name="description" content="Chapter 7 KEGG enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 KEGG enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 KEGG enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  
  
  



<meta name="date" content="2020-09-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="clusterprofiler-go.html"/>
<link rel="next" href="reactomepa.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.15/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>üìñ Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>üéØ Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i>üìù Citation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-structure"><i class="fa fa-check"></i>üìö Book structure</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#want-to-help"><i class="fa fa-check"></i>üíñ Want to help?</a></li>
</ul></li>
<li class="part"><span><b>Part I: Semantic Similarity Analysis</b></span></li>
<li class="chapter" data-level="1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html"><i class="fa fa-check"></i><b>1</b> Overview of semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="1.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#information-content-based-methods"><i class="fa fa-check"></i><b>1.1</b> Information content-based methods</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#resnik-method"><i class="fa fa-check"></i><b>1.1.1</b> Resnik method</a></li>
<li class="chapter" data-level="1.1.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#lin-method"><i class="fa fa-check"></i><b>1.1.2</b> Lin method</a></li>
<li class="chapter" data-level="1.1.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#rel-method"><i class="fa fa-check"></i><b>1.1.3</b> Rel method</a></li>
<li class="chapter" data-level="1.1.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#jiang-method"><i class="fa fa-check"></i><b>1.1.4</b> Jiang method</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#graph-based-method"><i class="fa fa-check"></i><b>1.2</b> Graph-based method</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#wang-method"><i class="fa fa-check"></i><b>1.2.1</b> Wang method</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#combine-methods"><i class="fa fa-check"></i><b>1.3</b> Combine methods</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#max"><i class="fa fa-check"></i><b>1.3.1</b> max</a></li>
<li class="chapter" data-level="1.3.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#avg"><i class="fa fa-check"></i><b>1.3.2</b> avg</a></li>
<li class="chapter" data-level="1.3.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#rcmax"><i class="fa fa-check"></i><b>1.3.3</b> rcmax</a></li>
<li class="chapter" data-level="1.3.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#bma"><i class="fa fa-check"></i><b>1.3.4</b> BMA</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="GOSemSim.html"><a href="GOSemSim.html"><i class="fa fa-check"></i><b>2</b> GO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="GOSemSim.html"><a href="GOSemSim.html#semantic-data"><i class="fa fa-check"></i><b>2.1</b> Semantic data</a></li>
<li class="chapter" data-level="2.2" data-path="GOSemSim.html"><a href="GOSemSim.html#gosemsim-supported-organisms"><i class="fa fa-check"></i><b>2.2</b> Supported organisms</a></li>
<li class="chapter" data-level="2.3" data-path="GOSemSim.html"><a href="GOSemSim.html#go-semantic-simiarlity"><i class="fa fa-check"></i><b>2.3</b> GO semantic similarity measurement</a></li>
<li class="chapter" data-level="2.4" data-path="GOSemSim.html"><a href="GOSemSim.html#gene-go-semantic-similarity"><i class="fa fa-check"></i><b>2.4</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="2.5" data-path="GOSemSim.html"><a href="GOSemSim.html#gene-cluster-go-semantic-similarity"><i class="fa fa-check"></i><b>2.5</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html"><i class="fa fa-check"></i><b>3</b> DO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#do-semantic-similarity"><i class="fa fa-check"></i><b>3.1</b> DO term semantic similarity measurement</a></li>
<li class="chapter" data-level="3.2" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#gene-do-semantic-similarity"><i class="fa fa-check"></i><b>3.2</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="3.3" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#gene-cluster-do-semantic-similarity"><i class="fa fa-check"></i><b>3.3</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html"><i class="fa fa-check"></i><b>4</b> MeSH semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#meshes-supported-organisms"><i class="fa fa-check"></i><b>4.1</b> Supported organisms</a></li>
<li class="chapter" data-level="4.2" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#meshes-semantic-simiarlity"><i class="fa fa-check"></i><b>4.2</b> MeSH semantic similarity measurement</a></li>
<li class="chapter" data-level="4.3" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#gene-meshes-semantic-similarity"><i class="fa fa-check"></i><b>4.3</b> Gene semantic similarity measurement</a></li>
</ul></li>
<li class="part"><span><b>Part II: Enrichment analysis</b></span></li>
<li class="chapter" data-level="5" data-path="enrichment-overview.html"><a href="enrichment-overview.html"><i class="fa fa-check"></i><b>5</b> Overview of enrichment analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="enrichment-overview.html"><a href="enrichment-overview.html#terminology"><i class="fa fa-check"></i><b>5.1</b> Terminology</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>5.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="5.1.2" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-ontology-go"><i class="fa fa-check"></i><b>5.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="5.1.3" data-path="enrichment-overview.html"><a href="enrichment-overview.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>5.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="5.1.4" data-path="enrichment-overview.html"><a href="enrichment-overview.html#other-gene-sets"><i class="fa fa-check"></i><b>5.1.4</b> Other gene sets</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="enrichment-overview.html"><a href="enrichment-overview.html#ora-algorithm"><i class="fa fa-check"></i><b>5.2</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="5.3" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gsea-algorithm"><i class="fa fa-check"></i><b>5.3</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="enrichment-overview.html"><a href="enrichment-overview.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>5.4</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html"><i class="fa fa-check"></i><b>6</b> GO enrichment analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterProfiler-go-supported-organisms"><i class="fa fa-check"></i><b>6.1</b> Supported organisms</a></li>
<li class="chapter" data-level="6.2" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-classification"><i class="fa fa-check"></i><b>6.2</b> GO classification</a></li>
<li class="chapter" data-level="6.3" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-ora"><i class="fa fa-check"></i><b>6.3</b> GO over-representation test</a></li>
<li class="chapter" data-level="6.4" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.5" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-non-model"><i class="fa fa-check"></i><b>6.5</b> GO analysis for non-model organisms</a></li>
<li class="chapter" data-level="6.6" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-summary"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html"><i class="fa fa-check"></i><b>7</b> KEGG enrichment analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterProfiler-kegg-supported-organisms"><i class="fa fa-check"></i><b>7.1</b> Supported organisms</a></li>
<li class="chapter" data-level="7.2" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-pathway-ora"><i class="fa fa-check"></i><b>7.2</b> KEGG pathway over-representation test</a></li>
<li class="chapter" data-level="7.3" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-pathway-gsea"><i class="fa fa-check"></i><b>7.3</b> KEGG pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="7.4" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-module-ora"><i class="fa fa-check"></i><b>7.4</b> KEGG module over-representation test</a></li>
<li class="chapter" data-level="7.5" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-module-gsea"><i class="fa fa-check"></i><b>7.5</b> KEGG module gene set enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reactomepa.html"><a href="reactomepa.html"><i class="fa fa-check"></i><b>8</b> Reactome enrichment analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-supported-organisms"><i class="fa fa-check"></i><b>8.1</b> Supported organisms</a></li>
<li class="chapter" data-level="8.2" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-ora"><i class="fa fa-check"></i><b>8.2</b> Reactome pathway over-representation test</a></li>
<li class="chapter" data-level="8.3" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-gsea"><i class="fa fa-check"></i><b>8.3</b> Reactome pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="8.4" data-path="reactomepa.html"><a href="reactomepa.html#pathway-visualization"><i class="fa fa-check"></i><b>8.4</b> Pathway Visualization</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dose-enrichment.html"><a href="dose-enrichment.html"><i class="fa fa-check"></i><b>9</b> DO enrichment analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="dose-enrichment.html"><a href="dose-enrichment.html#enrichdo-function"><i class="fa fa-check"></i><b>9.1</b> <code>enrichDO</code> function</a></li>
<li class="chapter" data-level="9.2" data-path="dose-enrichment.html"><a href="dose-enrichment.html#enrichncg-function"><i class="fa fa-check"></i><b>9.2</b> <code>enrichNCG</code> function</a></li>
<li class="chapter" data-level="9.3" data-path="dose-enrichment.html"><a href="dose-enrichment.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>9.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
<li class="chapter" data-level="9.4" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsedo-fuction"><i class="fa fa-check"></i><b>9.4</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="9.5" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsencg-fuction"><i class="fa fa-check"></i><b>9.5</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="9.6" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsedgn-fuction"><i class="fa fa-check"></i><b>9.6</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="meshes-enrichment.html"><a href="meshes-enrichment.html"><i class="fa fa-check"></i><b>10</b> MeSH enrichment analysis</a></li>
<li class="chapter" data-level="11" data-path="universal-api.html"><a href="universal-api.html"><i class="fa fa-check"></i><b>11</b> Universal enrichment analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="universal-api.html"><a href="universal-api.html#input-data"><i class="fa fa-check"></i><b>11.1</b> Input data</a></li>
<li class="chapter" data-level="11.2" data-path="universal-api.html"><a href="universal-api.html#wikipathways-analysis"><i class="fa fa-check"></i><b>11.2</b> WikiPathways analysis</a></li>
<li class="chapter" data-level="11.3" data-path="universal-api.html"><a href="universal-api.html#cell-marker"><i class="fa fa-check"></i><b>11.3</b> Cell Marker</a></li>
<li class="chapter" data-level="11.4" data-path="universal-api.html"><a href="universal-api.html#msigdb-analysis"><i class="fa fa-check"></i><b>11.4</b> MSigDb analysis</a></li>
<li class="chapter" data-level="11.5" data-path="universal-api.html"><a href="universal-api.html#msigdb"><i class="fa fa-check"></i><b>11.5</b> MSigDb</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chipseeker-enrichment.html"><a href="chipseeker-enrichment.html"><i class="fa fa-check"></i><b>12</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="13" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html"><i class="fa fa-check"></i><b>13</b> Biological theme comparison</a>
<ul>
<li class="chapter" data-level="13.1" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>13.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="13.2" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>13.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="enrichplot.html"><a href="enrichplot.html"><i class="fa fa-check"></i><b>14</b> Visualization of functional enrichment result</a>
<ul>
<li class="chapter" data-level="14.1" data-path="enrichplot.html"><a href="enrichplot.html#bar-plot"><i class="fa fa-check"></i><b>14.1</b> Bar Plot</a></li>
<li class="chapter" data-level="14.2" data-path="enrichplot.html"><a href="enrichplot.html#dot-plot"><i class="fa fa-check"></i><b>14.2</b> Dot plot</a></li>
<li class="chapter" data-level="14.3" data-path="enrichplot.html"><a href="enrichplot.html#gene-concept-network"><i class="fa fa-check"></i><b>14.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="14.4" data-path="enrichplot.html"><a href="enrichplot.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>14.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="14.5" data-path="enrichplot.html"><a href="enrichplot.html#enrichment-map"><i class="fa fa-check"></i><b>14.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="14.6" data-path="enrichplot.html"><a href="enrichplot.html#upset-plot"><i class="fa fa-check"></i><b>14.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="14.7" data-path="enrichplot.html"><a href="enrichplot.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>14.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="14.8" data-path="enrichplot.html"><a href="enrichplot.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>14.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="14.9" data-path="enrichplot.html"><a href="enrichplot.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>14.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="14.10" data-path="enrichplot.html"><a href="enrichplot.html#goplot"><i class="fa fa-check"></i><b>14.10</b> goplot</a></li>
<li class="chapter" data-level="14.11" data-path="enrichplot.html"><a href="enrichplot.html#browsekegg"><i class="fa fa-check"></i><b>14.11</b> browseKEGG</a></li>
<li class="chapter" data-level="14.12" data-path="enrichplot.html"><a href="enrichplot.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>14.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="15" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html"><i class="fa fa-check"></i><b>15</b> dplyr verbs for manipulating enrichment result</a>
<ul>
<li class="chapter" data-level="15.1" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#filter"><i class="fa fa-check"></i><b>15.1</b> filter</a></li>
<li class="chapter" data-level="15.2" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#arrange"><i class="fa fa-check"></i><b>15.2</b> arrange</a></li>
<li class="chapter" data-level="15.3" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#select"><i class="fa fa-check"></i><b>15.3</b> select</a></li>
<li class="chapter" data-level="15.4" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#mutate"><i class="fa fa-check"></i><b>15.4</b> mutate</a></li>
<li class="chapter" data-level="15.5" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#slice"><i class="fa fa-check"></i><b>15.5</b> slice</a></li>
<li class="chapter" data-level="15.6" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#summarise"><i class="fa fa-check"></i><b>15.6</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="useful-utilities.html"><a href="useful-utilities.html"><i class="fa fa-check"></i><b>16</b> Useful utilities</a>
<ul>
<li class="chapter" data-level="16.1" data-path="useful-utilities.html"><a href="useful-utilities.html#id-convert"><i class="fa fa-check"></i><b>16.1</b> Translating Biological ID</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="useful-utilities.html"><a href="useful-utilities.html#bitr"><i class="fa fa-check"></i><b>16.1.1</b> bitr: Biological Id TranslatoR</a></li>
<li class="chapter" data-level="16.1.2" data-path="useful-utilities.html"><a href="useful-utilities.html#bitr_kegg"><i class="fa fa-check"></i><b>16.1.2</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="useful-utilities.html"><a href="useful-utilities.html#setReadable"><i class="fa fa-check"></i><b>16.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a>
<ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#genelist"><i class="fa fa-check"></i><b>A.1</b> How to prepare your own geneList</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clusterprofiler-kegg" class="section level1" number="7">
<h1><span class="header-section-number">Chapter 7</span> KEGG enrichment analysis</h1>
<p>The KEGG FTP service is not freely available for academic since 2012, and there are many software packages using out-dated KEGG annotation data. The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package supports downloading latest online version of KEGG data using the <a href="https://www.kegg.jp">KEGG website</a>, which is freely available for academic users. Both KEGG pathway and module are supported in <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>.</p>
<div id="clusterProfiler-kegg-supported-organisms" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Supported organisms</h2>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package supports all organisms that have KEGG annotation data available in KEGG database. User should pass abbreviation of academic name to the <code>organism</code> parameter. The full list of KEGG supported organisms can be accessed via <a href="http://www.genome.jp/kegg/catalog/org_list.html" class="uri">http://www.genome.jp/kegg/catalog/org_list.html</a>. <a href="https://www.genome.jp/kegg/ko.html">KEGG Orthology</a> (KO) Database is also supported by specifying <code>organism = "ko"</code>.</p>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides <code>search_kegg_organism()</code> function to help searching supported organisms.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="clusterprofiler-kegg.html#cb68-1" aria-hidden="true"></a><span class="kw">library</span>(clusterProfiler)</span>
<span id="cb68-2"><a href="clusterprofiler-kegg.html#cb68-2" aria-hidden="true"></a><span class="kw">search_kegg_organism</span>(<span class="st">&#39;ece&#39;</span>, <span class="dt">by=</span><span class="st">&#39;kegg_code&#39;</span>)</span></code></pre></div>
<pre><code>##     kegg_code                        scientific_name common_name
## 366       ece Escherichia coli O157:H7 EDL933 (EHEC)        &lt;NA&gt;</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="clusterprofiler-kegg.html#cb70-1" aria-hidden="true"></a>ecoli &lt;-<span class="st"> </span><span class="kw">search_kegg_organism</span>(<span class="st">&#39;Escherichia coli&#39;</span>, <span class="dt">by=</span><span class="st">&#39;scientific_name&#39;</span>)</span>
<span id="cb70-2"><a href="clusterprofiler-kegg.html#cb70-2" aria-hidden="true"></a><span class="kw">dim</span>(ecoli)</span></code></pre></div>
<pre><code>## [1] 65  3</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="clusterprofiler-kegg.html#cb72-1" aria-hidden="true"></a><span class="kw">head</span>(ecoli)</span></code></pre></div>
<pre><code>##     kegg_code                        scientific_name common_name
## 361       eco           Escherichia coli K-12 MG1655        &lt;NA&gt;
## 362       ecj            Escherichia coli K-12 W3110        &lt;NA&gt;
## 363       ecd            Escherichia coli K-12 DH10B        &lt;NA&gt;
## 364       ebw                Escherichia coli BW2952        &lt;NA&gt;
## 365      ecok            Escherichia coli K-12 MDS42        &lt;NA&gt;
## 366       ece Escherichia coli O157:H7 EDL933 (EHEC)        &lt;NA&gt;</code></pre>
</div>
<div id="clusterprofiler-kegg-pathway-ora" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> KEGG pathway over-representation test</h2>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="clusterprofiler-kegg.html#cb74-1" aria-hidden="true"></a><span class="kw">data</span>(geneList, <span class="dt">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb74-2"><a href="clusterprofiler-kegg.html#cb74-2" aria-hidden="true"></a>gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]</span>
<span id="cb74-3"><a href="clusterprofiler-kegg.html#cb74-3" aria-hidden="true"></a></span>
<span id="cb74-4"><a href="clusterprofiler-kegg.html#cb74-4" aria-hidden="true"></a>kk &lt;-<span class="st"> </span><span class="kw">enrichKEGG</span>(<span class="dt">gene         =</span> gene,</span>
<span id="cb74-5"><a href="clusterprofiler-kegg.html#cb74-5" aria-hidden="true"></a>                 <span class="dt">organism     =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb74-6"><a href="clusterprofiler-kegg.html#cb74-6" aria-hidden="true"></a>                 <span class="dt">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb74-7"><a href="clusterprofiler-kegg.html#cb74-7" aria-hidden="true"></a><span class="kw">head</span>(kk)</span></code></pre></div>
<pre><code>##                ID                                                   Description
## hsa04110 hsa04110                                                    Cell cycle
## hsa04114 hsa04114                                                Oocyte meiosis
## hsa04218 hsa04218                                           Cellular senescence
## hsa04061 hsa04061 Viral protein interaction with cytokine and cytokine receptor
## hsa03320 hsa03320                                        PPAR signaling pathway
## hsa04914 hsa04914                       Progesterone-mediated oocyte maturation
##          GeneRatio  BgRatio       pvalue     p.adjust       qvalue
## hsa04110     11/94 124/8063 1.709528e-07 3.453247e-05 3.365071e-05
## hsa04114     10/94 128/8063 2.089725e-06 2.110622e-04 2.056729e-04
## hsa04218     10/94 156/8063 1.235106e-05 8.316384e-04 8.104032e-04
## hsa04061      8/94 100/8063 1.952541e-05 9.860331e-04 9.608556e-04
## hsa03320      7/94  77/8063 2.891619e-05 1.168214e-03 1.138385e-03
## hsa04914      7/94  99/8063 1.453968e-04 4.895026e-03 4.770036e-03
##                                                      geneID Count
## hsa04110 8318/991/9133/890/983/4085/7272/1111/891/4174/9232    11
## hsa04114    991/9133/983/4085/51806/6790/891/9232/3708/5241    10
## hsa04218     2305/4605/9133/890/983/51806/1111/891/776/3708    10
## hsa04061           3627/10563/6373/4283/6362/6355/9547/1524     8
## hsa03320                 4312/9415/9370/5105/2167/3158/5346     7
## hsa04914                    9133/890/983/4085/6790/891/5241     7</code></pre>
<p>Input ID type can be <code>kegg</code>, <code>ncbi-geneid</code>, <code>ncbi-proteinid</code> or <code>uniprot</code> (see also <a href="useful-utilities.html#bitr_kegg">session 16.1.2</a>). Unlike <code>enrichGO()</code>, there is no <code>readable</code> parameter for <code>enrichKEGG()</code>. However, users can use the <a href="useful-utilities.html#setReadable"><code>setReadable()</code> function</a> if there is an <code>OrgDb</code> available for the species.</p>
</div>
<div id="clusterprofiler-kegg-pathway-gsea" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> KEGG pathway gene set enrichment analysis</h2>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="clusterprofiler-kegg.html#cb76-1" aria-hidden="true"></a>kk2 &lt;-<span class="st"> </span><span class="kw">gseKEGG</span>(<span class="dt">geneList     =</span> geneList,</span>
<span id="cb76-2"><a href="clusterprofiler-kegg.html#cb76-2" aria-hidden="true"></a>               <span class="dt">organism     =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb76-3"><a href="clusterprofiler-kegg.html#cb76-3" aria-hidden="true"></a>               <span class="dt">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb76-4"><a href="clusterprofiler-kegg.html#cb76-4" aria-hidden="true"></a>               <span class="dt">minGSSize    =</span> <span class="dv">120</span>,</span>
<span id="cb76-5"><a href="clusterprofiler-kegg.html#cb76-5" aria-hidden="true"></a>               <span class="dt">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb76-6"><a href="clusterprofiler-kegg.html#cb76-6" aria-hidden="true"></a>               <span class="dt">verbose      =</span> <span class="ot">FALSE</span>)</span>
<span id="cb76-7"><a href="clusterprofiler-kegg.html#cb76-7" aria-hidden="true"></a><span class="kw">head</span>(kk2)</span></code></pre></div>
<pre><code>##                ID                Description setSize enrichmentScore       NES
## hsa04151 hsa04151 PI3K-Akt signaling pathway     322      -0.3482755 -1.500870
## hsa04510 hsa04510             Focal adhesion     190      -0.4169068 -1.704878
## hsa03013 hsa03013              RNA transport     135       0.4103684  1.754986
## hsa04218 hsa04218        Cellular senescence     141       0.4115945  1.773355
## hsa05152 hsa05152               Tuberculosis     162       0.3745153  1.652657
## hsa05203 hsa05203       Viral carcinogenesis     164       0.3523856  1.557636
##               pvalue   p.adjust    qvalues rank                   leading_edge
## hsa04151 0.001277139 0.02734375 0.01768092 1997 tags=23%, list=16%, signal=20%
## hsa04510 0.001356852 0.02734375 0.01768092 2183 tags=27%, list=17%, signal=22%
## hsa03013 0.003278689 0.02734375 0.01768092 3383 tags=39%, list=27%, signal=29%
## hsa04218 0.003289474 0.02734375 0.01768092 1155  tags=17%, list=9%, signal=16%
## hsa05152 0.003355705 0.02734375 0.01768092 2823 tags=34%, list=23%, signal=27%
## hsa05203 0.003367003 0.02734375 0.01768092 3112 tags=35%, list=25%, signal=26%
##                                                                                                                                                                                                                                                                                                                                                                                     core_enrichment
## hsa04151 627/2252/7059/92579/5563/5295/6794/1288/7010/3910/3371/3082/1291/4602/3791/1027/90993/3441/3643/1129/2322/1975/7450/596/3685/1942/2149/1280/4804/3675/595/2261/7248/2246/4803/3912/1902/1278/1277/2846/2057/1293/2247/55970/5618/7058/10161/56034/3693/4254/3480/4908/5159/1292/3908/2690/3909/8817/9223/4915/3551/2791/63923/3913/9863/3667/1287/3679/7060/3479/80310/1311/5105/2066/1101
## hsa04510                                                                                                                        5595/5228/7424/1499/4636/83660/7059/5295/1288/23396/3910/3371/3082/1291/394/3791/7450/596/3685/1280/3675/595/2318/3912/1793/1278/1277/1293/10398/55742/2317/7058/25759/56034/3693/3480/5159/857/1292/3908/3909/63923/3913/1287/3679/7060/3479/10451/80310/1311/1101
## hsa03013                                                                                       10460/1978/55110/54913/9688/8894/11260/10799/9631/4116/5042/8761/6396/23165/8662/10248/55706/79833/9775/29107/23636/5905/9513/5901/10775/10557/4927/79902/1981/26986/11171/10762/8480/8891/11097/26019/55308/10940/4686/9972/81929/10556/3646/9470/387082/1977/57122/8563/7514/79023/3837/9818/56000
## hsa04218                                                                                                                                                                                                                                                                      2305/4605/9133/890/983/51806/1111/891/993/3576/1978/898/9134/4609/1869/1029/22808/1871/5499/91860/292/1019/11200/1875
## hsa05152                                                                                                      820/51806/6772/64581/3126/3112/8767/3654/1054/1051/3458/1520/11151/1594/50617/54205/91860/8877/3329/637/3689/7096/2207/3929/4360/5603/929/533/3452/6850/7124/1509/3569/7097/1378/8772/64170/3119/843/2213/8625/3920/2215/3587/5594/3593/9103/3592/6300/9114/10333/3109/3108/1432/3552
## hsa05203                                                                                                991/890/983/1111/898/9134/6502/85236/1237/1029/8970/4067/2957/5902/55697/3066/578/1019/1234/8345/7534/5315/3018/468/6850/1030/4055/7186/1959/3135/5933/5610/1642/5594/9734/4792/5567/3134/23352/7185/10379/5366/4893/1232/2960/23513/9114/2958/10971/7533/9519/5568/836/1233/6714/3106/8379</code></pre>
</div>
<div id="clusterprofiler-kegg-module-ora" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> KEGG module over-representation test</h2>
<p><a href="http://www.genome.jp/kegg/module.html">KEGG Module</a> is a collection of manually defined function units. In some situation, KEGG Modules have a more straightforward interpretation.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="clusterprofiler-kegg.html#cb78-1" aria-hidden="true"></a>mkk &lt;-<span class="st"> </span><span class="kw">enrichMKEGG</span>(<span class="dt">gene =</span> gene,</span>
<span id="cb78-2"><a href="clusterprofiler-kegg.html#cb78-2" aria-hidden="true"></a>                   <span class="dt">organism =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb78-3"><a href="clusterprofiler-kegg.html#cb78-3" aria-hidden="true"></a>                   <span class="dt">pvalueCutoff =</span> <span class="dv">1</span>,</span>
<span id="cb78-4"><a href="clusterprofiler-kegg.html#cb78-4" aria-hidden="true"></a>                   <span class="dt">qvalueCutoff =</span> <span class="dv">1</span>)</span>
<span id="cb78-5"><a href="clusterprofiler-kegg.html#cb78-5" aria-hidden="true"></a><span class="kw">head</span>(mkk)                   </span></code></pre></div>
<pre><code>##            ID
## M00095 M00095
## M00053 M00053
## M00003 M00003
## M00049 M00049
##                                                                       Description
## M00095                             C5 isoprenoid biosynthesis, mevalonate pathway
## M00053 Pyrimidine deoxyribonuleotide biosynthesis, CDP/CTP =&gt; dCDP/dCTP,dTDP/dTTP
## M00003                               Gluconeogenesis, oxaloacetate =&gt; fructose-6P
## M00049                        Adenine ribonucleotide biosynthesis, IMP =&gt; ADP,ATP
##        GeneRatio BgRatio    pvalue  p.adjust    qvalue geneID Count
## M00095       1/9  10/824 0.1045527 0.2171109 0.2171109   3158     1
## M00053       1/9  14/824 0.1435736 0.2171109 0.2171109   6241     1
## M00003       1/9  18/824 0.1810757 0.2171109 0.2171109   5105     1
## M00049       1/9  22/824 0.2171109 0.2171109 0.2171109  26289     1</code></pre>
</div>
<div id="clusterprofiler-kegg-module-gsea" class="section level2" number="7.5">
<h2><span class="header-section-number">7.5</span> KEGG module gene set enrichment analysis</h2>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="clusterprofiler-kegg.html#cb80-1" aria-hidden="true"></a>mkk2 &lt;-<span class="st"> </span><span class="kw">gseMKEGG</span>(<span class="dt">geneList =</span> geneList,</span>
<span id="cb80-2"><a href="clusterprofiler-kegg.html#cb80-2" aria-hidden="true"></a>                 <span class="dt">organism =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb80-3"><a href="clusterprofiler-kegg.html#cb80-3" aria-hidden="true"></a>                 <span class="dt">pvalueCutoff =</span> <span class="dv">1</span>)</span>
<span id="cb80-4"><a href="clusterprofiler-kegg.html#cb80-4" aria-hidden="true"></a><span class="kw">head</span>(mkk2)</span></code></pre></div>
<pre><code>##            ID
## M00001 M00001
## M00035 M00035
## M00053 M00053
## M00002 M00002
## M00009 M00009
## M00104 M00104
##                                                                       Description
## M00001                  Glycolysis (Embden-Meyerhof pathway), glucose =&gt; pyruvate
## M00035                                                     Methionine degradation
## M00053 Pyrimidine deoxyribonuleotide biosynthesis, CDP/CTP =&gt; dCDP/dCTP,dTDP/dTTP
## M00002                   Glycolysis, core module involving three-carbon compounds
## M00009                                     Citrate cycle (TCA cycle, Krebs cycle)
## M00104           Bile acid biosynthesis, cholesterol =&gt; cholate/chenodeoxycholate
##        setSize enrichmentScore       NES      pvalue  p.adjust   qvalues rank
## M00001      24       0.5739036  1.761163 0.007506083 0.2401947 0.2212319 2886
## M00035      10       0.6784636  1.635711 0.018802441 0.2862329 0.2636356 1555
## M00053      10       0.6648004  1.602770 0.026834338 0.2862329 0.2636356  648
## M00002      11       0.6421781  1.594902 0.036559118 0.2924729 0.2693830 1381
## M00009      22       0.4504911  1.347220 0.118518519 0.7389091 0.6805742 3514
## M00104      10      -0.5876900 -1.323981 0.140221402 0.7389091 0.6805742  961
##                          leading_edge
## M00001 tags=54%, list=23%, signal=42%
## M00035 tags=50%, list=12%, signal=44%
## M00053  tags=40%, list=5%, signal=38%
## M00002 tags=55%, list=11%, signal=49%
## M00009 tags=50%, list=28%, signal=36%
## M00104  tags=50%, list=8%, signal=46%
##                                                         core_enrichment
## M00001 5214/3101/2821/7167/2597/5230/2023/5223/5315/3099/5232/2027/5211
## M00035                                           875/1789/191/1788/1786
## M00053                                              6241/7298/4830/1841
## M00002                                    7167/2597/5230/2023/5223/5315
## M00009            3418/50/4190/3419/2271/3421/55753/3417/1431/6389/4191
## M00104                                        6342/10998/1581/3295/8309</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="clusterprofiler-go.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="reactomepa.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
