<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Reactome enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
  <meta name="description" content="Chapter 8 Reactome enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Reactome enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Reactome enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  
  
  



<meta name="date" content="2020-09-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="clusterprofiler-kegg.html"/>
<link rel="next" href="dose-enrichment.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.15/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>üìñ Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>üéØ Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i>üìù Citation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-structure"><i class="fa fa-check"></i>üìö Book structure</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#want-to-help"><i class="fa fa-check"></i>üíñ Want to help?</a></li>
</ul></li>
<li class="part"><span><b>Part I: Semantic Similarity Analysis</b></span></li>
<li class="chapter" data-level="1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html"><i class="fa fa-check"></i><b>1</b> Overview of semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="1.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#information-content-based-methods"><i class="fa fa-check"></i><b>1.1</b> Information content-based methods</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#resnik-method"><i class="fa fa-check"></i><b>1.1.1</b> Resnik method</a></li>
<li class="chapter" data-level="1.1.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#lin-method"><i class="fa fa-check"></i><b>1.1.2</b> Lin method</a></li>
<li class="chapter" data-level="1.1.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#rel-method"><i class="fa fa-check"></i><b>1.1.3</b> Rel method</a></li>
<li class="chapter" data-level="1.1.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#jiang-method"><i class="fa fa-check"></i><b>1.1.4</b> Jiang method</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#graph-based-method"><i class="fa fa-check"></i><b>1.2</b> Graph-based method</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#wang-method"><i class="fa fa-check"></i><b>1.2.1</b> Wang method</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#combine-methods"><i class="fa fa-check"></i><b>1.3</b> Combine methods</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#max"><i class="fa fa-check"></i><b>1.3.1</b> max</a></li>
<li class="chapter" data-level="1.3.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#avg"><i class="fa fa-check"></i><b>1.3.2</b> avg</a></li>
<li class="chapter" data-level="1.3.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#rcmax"><i class="fa fa-check"></i><b>1.3.3</b> rcmax</a></li>
<li class="chapter" data-level="1.3.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#bma"><i class="fa fa-check"></i><b>1.3.4</b> BMA</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="GOSemSim.html"><a href="GOSemSim.html"><i class="fa fa-check"></i><b>2</b> GO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="GOSemSim.html"><a href="GOSemSim.html#semantic-data"><i class="fa fa-check"></i><b>2.1</b> Semantic data</a></li>
<li class="chapter" data-level="2.2" data-path="GOSemSim.html"><a href="GOSemSim.html#gosemsim-supported-organisms"><i class="fa fa-check"></i><b>2.2</b> Supported organisms</a></li>
<li class="chapter" data-level="2.3" data-path="GOSemSim.html"><a href="GOSemSim.html#go-semantic-simiarlity"><i class="fa fa-check"></i><b>2.3</b> GO semantic similarity measurement</a></li>
<li class="chapter" data-level="2.4" data-path="GOSemSim.html"><a href="GOSemSim.html#gene-go-semantic-similarity"><i class="fa fa-check"></i><b>2.4</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="2.5" data-path="GOSemSim.html"><a href="GOSemSim.html#gene-cluster-go-semantic-similarity"><i class="fa fa-check"></i><b>2.5</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html"><i class="fa fa-check"></i><b>3</b> DO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#do-semantic-similarity"><i class="fa fa-check"></i><b>3.1</b> DO term semantic similarity measurement</a></li>
<li class="chapter" data-level="3.2" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#gene-do-semantic-similarity"><i class="fa fa-check"></i><b>3.2</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="3.3" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#gene-cluster-do-semantic-similarity"><i class="fa fa-check"></i><b>3.3</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html"><i class="fa fa-check"></i><b>4</b> MeSH semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#meshes-supported-organisms"><i class="fa fa-check"></i><b>4.1</b> Supported organisms</a></li>
<li class="chapter" data-level="4.2" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#meshes-semantic-simiarlity"><i class="fa fa-check"></i><b>4.2</b> MeSH semantic similarity measurement</a></li>
<li class="chapter" data-level="4.3" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#gene-meshes-semantic-similarity"><i class="fa fa-check"></i><b>4.3</b> Gene semantic similarity measurement</a></li>
</ul></li>
<li class="part"><span><b>Part II: Enrichment analysis</b></span></li>
<li class="chapter" data-level="5" data-path="enrichment-overview.html"><a href="enrichment-overview.html"><i class="fa fa-check"></i><b>5</b> Overview of enrichment analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="enrichment-overview.html"><a href="enrichment-overview.html#terminology"><i class="fa fa-check"></i><b>5.1</b> Terminology</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>5.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="5.1.2" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-ontology-go"><i class="fa fa-check"></i><b>5.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="5.1.3" data-path="enrichment-overview.html"><a href="enrichment-overview.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>5.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="5.1.4" data-path="enrichment-overview.html"><a href="enrichment-overview.html#other-gene-sets"><i class="fa fa-check"></i><b>5.1.4</b> Other gene sets</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="enrichment-overview.html"><a href="enrichment-overview.html#ora-algorithm"><i class="fa fa-check"></i><b>5.2</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="5.3" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gsea-algorithm"><i class="fa fa-check"></i><b>5.3</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="enrichment-overview.html"><a href="enrichment-overview.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>5.4</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html"><i class="fa fa-check"></i><b>6</b> GO enrichment analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterProfiler-go-supported-organisms"><i class="fa fa-check"></i><b>6.1</b> Supported organisms</a></li>
<li class="chapter" data-level="6.2" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-classification"><i class="fa fa-check"></i><b>6.2</b> GO classification</a></li>
<li class="chapter" data-level="6.3" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-ora"><i class="fa fa-check"></i><b>6.3</b> GO over-representation analysis</a></li>
<li class="chapter" data-level="6.4" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.5" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-non-model"><i class="fa fa-check"></i><b>6.5</b> GO analysis for non-model organisms</a></li>
<li class="chapter" data-level="6.6" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-summary"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html"><i class="fa fa-check"></i><b>7</b> KEGG enrichment analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterProfiler-kegg-supported-organisms"><i class="fa fa-check"></i><b>7.1</b> Supported organisms</a></li>
<li class="chapter" data-level="7.2" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-pathway-ora"><i class="fa fa-check"></i><b>7.2</b> KEGG pathway over-representation analysis</a></li>
<li class="chapter" data-level="7.3" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-pathway-gsea"><i class="fa fa-check"></i><b>7.3</b> KEGG pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="7.4" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-module-ora"><i class="fa fa-check"></i><b>7.4</b> KEGG module over-representation analysis</a></li>
<li class="chapter" data-level="7.5" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-module-gsea"><i class="fa fa-check"></i><b>7.5</b> KEGG module gene set enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reactomepa.html"><a href="reactomepa.html"><i class="fa fa-check"></i><b>8</b> Reactome enrichment analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-supported-organisms"><i class="fa fa-check"></i><b>8.1</b> Supported organisms</a></li>
<li class="chapter" data-level="8.2" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-ora"><i class="fa fa-check"></i><b>8.2</b> Reactome pathway over-representation analysis</a></li>
<li class="chapter" data-level="8.3" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-gsea"><i class="fa fa-check"></i><b>8.3</b> Reactome pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="8.4" data-path="reactomepa.html"><a href="reactomepa.html#pathway-visualization"><i class="fa fa-check"></i><b>8.4</b> Pathway Visualization</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dose-enrichment.html"><a href="dose-enrichment.html"><i class="fa fa-check"></i><b>9</b> Disease enrichment analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="dose-enrichment.html"><a href="dose-enrichment.html#dose-ora"><i class="fa fa-check"></i><b>9.1</b> Disease over-representation analysis</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="dose-enrichment.html"><a href="dose-enrichment.html#dose-do-ora"><i class="fa fa-check"></i><b>9.1.1</b> Over-representation analysis for disease ontology</a></li>
<li class="chapter" data-level="9.1.2" data-path="dose-enrichment.html"><a href="dose-enrichment.html#dose-ncg-ora"><i class="fa fa-check"></i><b>9.1.2</b> Over-representation analysis for the network of cancer gene</a></li>
<li class="chapter" data-level="9.1.3" data-path="dose-enrichment.html"><a href="dose-enrichment.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>9.1.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="dose-enrichment.html"><a href="dose-enrichment.html#dose-gsea"><i class="fa fa-check"></i><b>9.2</b> Disease gene set enrichment analysis</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsedo-fuction"><i class="fa fa-check"></i><b>9.2.1</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="9.2.2" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsencg-fuction"><i class="fa fa-check"></i><b>9.2.2</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="9.2.3" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsedgn-fuction"><i class="fa fa-check"></i><b>9.2.3</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="meshes-enrichment.html"><a href="meshes-enrichment.html"><i class="fa fa-check"></i><b>10</b> MeSH enrichment analysis</a></li>
<li class="chapter" data-level="11" data-path="universal-api.html"><a href="universal-api.html"><i class="fa fa-check"></i><b>11</b> Universal enrichment analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="universal-api.html"><a href="universal-api.html#input-data"><i class="fa fa-check"></i><b>11.1</b> Input data</a></li>
<li class="chapter" data-level="11.2" data-path="universal-api.html"><a href="universal-api.html#wikipathways-analysis"><i class="fa fa-check"></i><b>11.2</b> WikiPathways analysis</a></li>
<li class="chapter" data-level="11.3" data-path="universal-api.html"><a href="universal-api.html#cell-marker"><i class="fa fa-check"></i><b>11.3</b> Cell Marker</a></li>
<li class="chapter" data-level="11.4" data-path="universal-api.html"><a href="universal-api.html#msigdb-analysis"><i class="fa fa-check"></i><b>11.4</b> MSigDb analysis</a></li>
<li class="chapter" data-level="11.5" data-path="universal-api.html"><a href="universal-api.html#msigdb"><i class="fa fa-check"></i><b>11.5</b> MSigDb</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chipseeker-enrichment.html"><a href="chipseeker-enrichment.html"><i class="fa fa-check"></i><b>12</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="13" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html"><i class="fa fa-check"></i><b>13</b> Biological theme comparison</a>
<ul>
<li class="chapter" data-level="13.1" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>13.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="13.2" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>13.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="enrichplot.html"><a href="enrichplot.html"><i class="fa fa-check"></i><b>14</b> Visualization of functional enrichment result</a>
<ul>
<li class="chapter" data-level="14.1" data-path="enrichplot.html"><a href="enrichplot.html#bar-plot"><i class="fa fa-check"></i><b>14.1</b> Bar Plot</a></li>
<li class="chapter" data-level="14.2" data-path="enrichplot.html"><a href="enrichplot.html#dot-plot"><i class="fa fa-check"></i><b>14.2</b> Dot plot</a></li>
<li class="chapter" data-level="14.3" data-path="enrichplot.html"><a href="enrichplot.html#gene-concept-network"><i class="fa fa-check"></i><b>14.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="14.4" data-path="enrichplot.html"><a href="enrichplot.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>14.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="14.5" data-path="enrichplot.html"><a href="enrichplot.html#enrichment-map"><i class="fa fa-check"></i><b>14.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="14.6" data-path="enrichplot.html"><a href="enrichplot.html#upset-plot"><i class="fa fa-check"></i><b>14.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="14.7" data-path="enrichplot.html"><a href="enrichplot.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>14.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="14.8" data-path="enrichplot.html"><a href="enrichplot.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>14.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="14.9" data-path="enrichplot.html"><a href="enrichplot.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>14.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="14.10" data-path="enrichplot.html"><a href="enrichplot.html#goplot"><i class="fa fa-check"></i><b>14.10</b> goplot</a></li>
<li class="chapter" data-level="14.11" data-path="enrichplot.html"><a href="enrichplot.html#browsekegg"><i class="fa fa-check"></i><b>14.11</b> browseKEGG</a></li>
<li class="chapter" data-level="14.12" data-path="enrichplot.html"><a href="enrichplot.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>14.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="15" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html"><i class="fa fa-check"></i><b>15</b> dplyr verbs for manipulating enrichment result</a>
<ul>
<li class="chapter" data-level="15.1" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#filter"><i class="fa fa-check"></i><b>15.1</b> filter</a></li>
<li class="chapter" data-level="15.2" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#arrange"><i class="fa fa-check"></i><b>15.2</b> arrange</a></li>
<li class="chapter" data-level="15.3" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#select"><i class="fa fa-check"></i><b>15.3</b> select</a></li>
<li class="chapter" data-level="15.4" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#mutate"><i class="fa fa-check"></i><b>15.4</b> mutate</a></li>
<li class="chapter" data-level="15.5" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#slice"><i class="fa fa-check"></i><b>15.5</b> slice</a></li>
<li class="chapter" data-level="15.6" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#summarise"><i class="fa fa-check"></i><b>15.6</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="useful-utilities.html"><a href="useful-utilities.html"><i class="fa fa-check"></i><b>16</b> Useful utilities</a>
<ul>
<li class="chapter" data-level="16.1" data-path="useful-utilities.html"><a href="useful-utilities.html#id-convert"><i class="fa fa-check"></i><b>16.1</b> Translating Biological ID</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="useful-utilities.html"><a href="useful-utilities.html#bitr"><i class="fa fa-check"></i><b>16.1.1</b> bitr: Biological Id TranslatoR</a></li>
<li class="chapter" data-level="16.1.2" data-path="useful-utilities.html"><a href="useful-utilities.html#bitr_kegg"><i class="fa fa-check"></i><b>16.1.2</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="useful-utilities.html"><a href="useful-utilities.html#setReadable"><i class="fa fa-check"></i><b>16.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a>
<ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#genelist"><i class="fa fa-check"></i><b>A.1</b> How to prepare your own geneList</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="reactomepa" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> Reactome enrichment analysis</h1>
<p><a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> is designed for reactome pathway based analysis <span class="citation">(Yu and He <a href="#ref-yu_reactomepa_2016" role="doc-biblioref">2016</a>)</span>. Reactome is an open-source, open access, manually curated and peer-reviewed pathway database.</p>
<div id="reactomepa-supported-organisms" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Supported organisms</h2>
<p>Currently <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> supports several model organisms, including ‚Äòcelegans‚Äô, ‚Äòfly‚Äô, ‚Äòhuman‚Äô, ‚Äòmouse‚Äô, ‚Äòrat‚Äô, ‚Äòyeast‚Äô and ‚Äòzebrafish‚Äô. The input gene ID should be Entrez gene ID. We recommend using <a href="useful-utilities.html#bitr"><code>clusterProfiler::bitr()</code></a> to convert biological IDs.</p>
</div>
<div id="reactomepa-ora" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Reactome pathway over-representation analysis</h2>
<p>Enrichment analysis is a widely used approach to identify biological
themes. <a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> implemented <code>enrichPathway()</code> that uses <a href="enrichment-overview.html#ora-algorithm">hypergeometric model</a> to assess whether the number of selected genes associated with a reactome pathway is larger than expected.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="reactomepa.html#cb82-1" aria-hidden="true"></a><span class="kw">library</span>(ReactomePA)</span>
<span id="cb82-2"><a href="reactomepa.html#cb82-2" aria-hidden="true"></a><span class="kw">data</span>(geneList)</span>
<span id="cb82-3"><a href="reactomepa.html#cb82-3" aria-hidden="true"></a>de &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="fl">1.5</span>]</span>
<span id="cb82-4"><a href="reactomepa.html#cb82-4" aria-hidden="true"></a><span class="kw">head</span>(de)</span></code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="reactomepa.html#cb84-1" aria-hidden="true"></a>x &lt;-<span class="st"> </span><span class="kw">enrichPathway</span>(<span class="dt">gene=</span>de, <span class="dt">pvalueCutoff =</span> <span class="fl">0.05</span>, <span class="dt">readable=</span><span class="ot">TRUE</span>)</span>
<span id="cb84-2"><a href="reactomepa.html#cb84-2" aria-hidden="true"></a><span class="kw">head</span>(x)</span></code></pre></div>
<pre><code>##                          ID
## R-HSA-69620     R-HSA-69620
## R-HSA-2500257 R-HSA-2500257
## R-HSA-141424   R-HSA-141424
## R-HSA-141444   R-HSA-141444
## R-HSA-69618     R-HSA-69618
## R-HSA-68877     R-HSA-68877
##                                                                                        Description
## R-HSA-69620                                                                 Cell Cycle Checkpoints
## R-HSA-2500257                                              Resolution of Sister Chromatid Cohesion
## R-HSA-141424                                         Amplification of signal from the kinetochores
## R-HSA-141444  Amplification  of signal from unattached  kinetochores via a MAD2  inhibitory signal
## R-HSA-69618                                                             Mitotic Spindle Checkpoint
## R-HSA-68877                                                                   Mitotic Prometaphase
##               GeneRatio   BgRatio       pvalue     p.adjust       qvalue
## R-HSA-69620      37/322 293/10654 9.962958e-14 7.173330e-11 6.355319e-11
## R-HSA-2500257    23/322 126/10654 2.824834e-12 1.016940e-09 9.009735e-10
## R-HSA-141424     20/322  96/10654 6.054136e-12 1.089744e-09 9.654753e-10
## R-HSA-141444     20/322  96/10654 6.054136e-12 1.089744e-09 9.654753e-10
## R-HSA-69618      21/322 112/10654 1.508315e-11 2.171973e-09 1.924292e-09
## R-HSA-68877      26/322 200/10654 2.966075e-10 3.559290e-08 3.153406e-08
##                                                                                                                                                                                                                                 geneID
## R-HSA-69620   CDC45/CDCA8/MCM10/CDC20/CENPE/CCNB2/NDC80/UBE2C/SKA1/CENPM/CENPN/CCNA2/CDK1/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/CHEK1/CCNB1/MCM5/MCM2/KIF2C/CDC25A/CDC6/PLK1/BUB1B/GTSE1/EXO1/ZWINT/CENPU/SPC25/CENPI/CCNE1/ORC6/ORC1/TAOK1
## R-HSA-2500257                                                                                CDCA8/CDC20/CENPE/CCNB2/NDC80/SKA1/CENPM/CENPN/CDK1/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/CCNB1/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-141424                                                                                                  CDCA8/CDC20/CENPE/NDC80/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-141444                                                                                                  CDCA8/CDC20/CENPE/NDC80/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-69618                                                                                             CDCA8/CDC20/CENPE/NDC80/UBE2C/SKA1/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/BIRC5/AURKB/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
## R-HSA-68877                                                                 CDCA8/CDC20/CENPE/CCNB2/NDC80/NCAPH/SKA1/NEK2/CENPM/CENPN/CDK1/ERCC6L/MAD2L1/KIF18A/BIRC5/NCAPG/AURKB/CCNB1/KIF2C/PLK1/BUB1B/ZWINT/CENPU/SPC25/CENPI/TAOK1
##               Count
## R-HSA-69620      37
## R-HSA-2500257    23
## R-HSA-141424     20
## R-HSA-141444     20
## R-HSA-69618      21
## R-HSA-68877      26</code></pre>
</div>
<div id="reactomepa-gsea" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Reactome pathway gene set enrichment analysis</h2>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="reactomepa.html#cb86-1" aria-hidden="true"></a>y &lt;-<span class="st"> </span><span class="kw">gsePathway</span>(geneList, <span class="dt">nPerm=</span><span class="dv">10000</span>,</span>
<span id="cb86-2"><a href="reactomepa.html#cb86-2" aria-hidden="true"></a>                <span class="dt">pvalueCutoff=</span><span class="fl">0.2</span>,</span>
<span id="cb86-3"><a href="reactomepa.html#cb86-3" aria-hidden="true"></a>                <span class="dt">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</span>
<span id="cb86-4"><a href="reactomepa.html#cb86-4" aria-hidden="true"></a><span class="kw">head</span>(y)</span></code></pre></div>
<pre><code>##                          ID
## R-HSA-1474244 R-HSA-1474244
## R-HSA-216083   R-HSA-216083
## R-HSA-3000178 R-HSA-3000178
## R-HSA-1650814 R-HSA-1650814
## R-HSA-3000171 R-HSA-3000171
## R-HSA-2022090 R-HSA-2022090
##                                                                Description
## R-HSA-1474244                            Extracellular matrix organization
## R-HSA-216083                            Integrin cell surface interactions
## R-HSA-3000178                                            ECM proteoglycans
## R-HSA-1650814                  Collagen biosynthesis and modifying enzymes
## R-HSA-3000171                       Non-integrin membrane-ECM interactions
## R-HSA-2022090 Assembly of collagen fibrils and other multimeric structures
##               setSize enrichmentScore       NES       pvalue    p.adjust
## R-HSA-1474244     266      -0.4576106 -1.936754 0.0001332268 0.003176563
## R-HSA-216083       80      -0.5123103 -1.860721 0.0001546790 0.003176563
## R-HSA-3000178      74      -0.6262504 -2.247901 0.0001555210 0.003176563
## R-HSA-1650814      53      -0.5915513 -2.000152 0.0001596934 0.003176563
## R-HSA-3000171      56      -0.5863352 -2.000599 0.0001598210 0.003176563
## R-HSA-2022090      54      -0.5600621 -1.898797 0.0001598977 0.003176563
##                   qvalues rank                   leading_edge
## R-HSA-1474244 0.002306638 1943 tags=33%, list=16%, signal=29%
## R-HSA-216083  0.002306638 1890 tags=39%, list=15%, signal=33%
## R-HSA-3000178 0.002306638 1890 tags=46%, list=15%, signal=39%
## R-HSA-1650814 0.002306638 1890 tags=47%, list=15%, signal=40%
## R-HSA-3000171 0.002306638 2538 tags=45%, list=20%, signal=36%
## R-HSA-2022090 0.002306638 1897 tags=54%, list=15%, signal=46%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                       core_enrichment
## R-HSA-1474244 825/8038/11132/4017/1288/4811/3910/3371/1291/3791/831/1301/4238/7450/3685/80781/1280/1306/4314/3675/8425/977/4054/7837/7042/3912/4322/1278/1511/4060/30008/1277/164656/22795/10516/81578/1293/2247/1295/58494/8076/5118/2192/1281/83700/50509/4319/1290/1513/11096/2202/4313/2199/3693/10536/1294/11117/3339/1462/1289/1292/3908/4016/3909/4053/6678/1296/633/5654/2331/63923/7043/3913/1300/2200/1634/7177/1287/3679/4680/2006/7373/1307/1311/1308/652/4148/54829/4239
## R-HSA-216083                                                                                                                                                                                                                                                                                                           1288/3371/1291/3791/7450/3685/80781/1280/3675/1278/4060/1277/1293/1295/58494/1281/83700/50509/1290/3693/1294/3339/1289/1292/1296/1300/2200/1287/3679/1307/1311
## R-HSA-3000178                                                                                                                                                                                                                                                                                             1288/3910/3371/1291/3685/1280/7042/3912/1278/4060/1277/1293/1281/50509/1290/3693/3339/1462/1289/1292/3908/3909/6678/633/2331/63923/7043/3913/1634/1287/3679/1311/4148/54829
## R-HSA-1650814                                                                                                                                                                                                                                                                                                                                        1288/1291/1301/80781/1280/1306/1278/1277/81578/1293/1295/5118/1281/50509/1290/10536/1294/1289/1292/1296/1300/1287/7373/1307/1308
## R-HSA-3000171                                                                                                                                                                                                                                                                                                                                           7057/3915/6385/4921/1288/3910/3371/1301/3685/1280/3912/1278/1277/2247/1281/50509/1290/3693/3339/1289/3908/3909/3913/1300/1287
## R-HSA-2022090                                                                                                                                                                                                                                                                                                                       4017/1288/1291/1301/80781/1280/1306/4314/977/7837/4322/1278/1277/1293/1295/5118/1281/50509/1290/1294/1289/1292/4016/3909/1296/1300/1287/7373/1308</code></pre>
</div>
<div id="pathway-visualization" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Pathway Visualization</h2>
<p><a href="http://bioconductor.org/packages/ReactomePA">ReactomePA</a> implemented the <code>viewPathway()</code> to visualized selected reactome pathway. More general purpose of visualization methods for ORA and GSEA results are provided in the <a href="http://bioconductor.org/packages/enrichplot">enrichplot</a> package and are documented on <a href="enrichplot.html#enrichplot">Chapter 14</a>.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="reactomepa.html#cb88-1" aria-hidden="true"></a><span class="kw">viewPathway</span>(<span class="st">&quot;E2F mediated regulation of DNA replication&quot;</span>, </span>
<span id="cb88-2"><a href="reactomepa.html#cb88-2" aria-hidden="true"></a>            <span class="dt">readable =</span> <span class="ot">TRUE</span>, </span>
<span id="cb88-3"><a href="reactomepa.html#cb88-3" aria-hidden="true"></a>            <span class="dt">foldChange =</span> geneList)</span></code></pre></div>
<p><img src="biomedicalKnowledge_files/figure-html/unnamed-chunk-24-1.png" width="768" /></p>
<!--

## Pathway analysis of NGS data

Pathway analysis using NGS data (eg, RNA-Seq and ChIP-Seq) can be performed by linking coding and non-coding regions to coding genes via [ChIPseeker](http://bioconductor.org/packages/ChIPseeker) package, which can annotates genomic regions to their nearest genes, host genes, and flanking genes respectivly. In addtion, it provides a function, __*seq2gene*__, that simultaneously considering host genes, promoter region and flanking gene from intergenic region that may under control via cis-regulation. This function maps genomic regions to genes in a many-to-many manner and facilitate functional analysis. For more details, please refer to [ChIPseeker](http://bioconductor.org/packages/ChIPseeker)[@yu_chipseeker_2015].


## Visualize enrichment result

We implement barplot, dotplot enrichment map and category-gene-network for visualization. It is very common to visualize the enrichment result in bar or pie chart. We believe the pie chart is misleading and only provide bar chart.
```r fig.height=6, fig.width=12}
barplot(x, showCategory=8)
```
```r fig.height=6, fig.width=12}
dotplot(x, showCategory=15)
```

Enrichment map can be viusalized by __*enrichMap*__:
```r fig.height=10, fig.width=10}
emapplot(x)
```



In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories, we developed __*cnetplot*__ function to extract the complex association between genes and diseases.
```r fig.height=8, fig.width=8}
cnetplot(x, categorySize="pvalue", foldChange=geneList)
```


## Visualize GSEA result

```r fig.height=8, fig.width=8}
emapplot(y, color="pvalue")
```


```r fig.height=7, fig.width=10}
gseaplot(y, geneSetID = "R-HSA-69242")
```


## Comparing enriched reactome pathways among gene clusters with clusterProfiler

We have developed an `R` package [clusterProfiler](http://bioconductor.org/packages/clusterProfiler)[@yu_clusterprofiler:_2012] for comparing biological themes among gene clusters. [ReactomePA](http://bioconductor.org/packages/ReactomePA) works fine with [clusterProfiler](http://bioconductor.org/packages/clusterProfiler) and can compare biological themes at reactome pathway perspective.

```r fig.height=8, fig.width=13, eval=FALSE}
require(clusterProfiler)
data(gcSample)
res <- compareCluster(gcSample, fun="enrichPathway")
dotplot(res)
```


-->

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-yu_reactomepa_2016">
<p>Yu, Guangchuang, and Qing-Yu He. 2016. ‚ÄúReactomePA: An R/Bioconductor Package for Reactome Pathway Analysis and Visualization.‚Äù <em>Molecular BioSystems</em> 12 (2): 477‚Äì79. <a href="https://doi.org/10.1039/C5MB00663E">https://doi.org/10.1039/C5MB00663E</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="clusterprofiler-kegg.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dose-enrichment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
