<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Overview of semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
  <meta name="description" content="Chapter 1 Overview of semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Overview of semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Overview of semantic similarity analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  
  
  



<meta name="date" content="2020-09-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="GOSemSim.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.15/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>üìñ Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>üéØ Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i>üìù Citation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-structure"><i class="fa fa-check"></i>üìö Book structure</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#want-to-help"><i class="fa fa-check"></i>üíñ Want to help?</a></li>
</ul></li>
<li class="part"><span><b>Part I: Semantic Similarity Analysis</b></span></li>
<li class="chapter" data-level="1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html"><i class="fa fa-check"></i><b>1</b> Overview of semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="1.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#information-content-based-methods"><i class="fa fa-check"></i><b>1.1</b> Information content-based methods</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#resnik-method"><i class="fa fa-check"></i><b>1.1.1</b> Resnik method</a></li>
<li class="chapter" data-level="1.1.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#lin-method"><i class="fa fa-check"></i><b>1.1.2</b> Lin method</a></li>
<li class="chapter" data-level="1.1.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#rel-method"><i class="fa fa-check"></i><b>1.1.3</b> Rel method</a></li>
<li class="chapter" data-level="1.1.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#jiang-method"><i class="fa fa-check"></i><b>1.1.4</b> Jiang method</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#graph-based-method"><i class="fa fa-check"></i><b>1.2</b> Graph-based method</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#wang-method"><i class="fa fa-check"></i><b>1.2.1</b> Wang method</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#combine-methods"><i class="fa fa-check"></i><b>1.3</b> Combine methods</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#max"><i class="fa fa-check"></i><b>1.3.1</b> max</a></li>
<li class="chapter" data-level="1.3.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#avg"><i class="fa fa-check"></i><b>1.3.2</b> avg</a></li>
<li class="chapter" data-level="1.3.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#rcmax"><i class="fa fa-check"></i><b>1.3.3</b> rcmax</a></li>
<li class="chapter" data-level="1.3.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#bma"><i class="fa fa-check"></i><b>1.3.4</b> BMA</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="GOSemSim.html"><a href="GOSemSim.html"><i class="fa fa-check"></i><b>2</b> GO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="GOSemSim.html"><a href="GOSemSim.html#semantic-data"><i class="fa fa-check"></i><b>2.1</b> Semantic data</a></li>
<li class="chapter" data-level="2.2" data-path="GOSemSim.html"><a href="GOSemSim.html#gosemsim-supported-organisms"><i class="fa fa-check"></i><b>2.2</b> Supported organisms</a></li>
<li class="chapter" data-level="2.3" data-path="GOSemSim.html"><a href="GOSemSim.html#go-semantic-simiarlity"><i class="fa fa-check"></i><b>2.3</b> GO semantic similarity measurement</a></li>
<li class="chapter" data-level="2.4" data-path="GOSemSim.html"><a href="GOSemSim.html#gene-go-semantic-similarity"><i class="fa fa-check"></i><b>2.4</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="2.5" data-path="GOSemSim.html"><a href="GOSemSim.html#gene-cluster-go-semantic-similarity"><i class="fa fa-check"></i><b>2.5</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html"><i class="fa fa-check"></i><b>3</b> DO semantic similarity analysis</a></li>
<li class="chapter" data-level="4" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html"><i class="fa fa-check"></i><b>4</b> MeSH semantic similarity analysis</a></li>
<li class="part"><span><b>Part II: Enrichment analysis</b></span></li>
<li class="chapter" data-level="5" data-path="enrichment-overview.html"><a href="enrichment-overview.html"><i class="fa fa-check"></i><b>5</b> Overview of enrichment analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="enrichment-overview.html"><a href="enrichment-overview.html#terminology"><i class="fa fa-check"></i><b>5.1</b> Terminology</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>5.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="5.1.2" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-ontology-go"><i class="fa fa-check"></i><b>5.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="5.1.3" data-path="enrichment-overview.html"><a href="enrichment-overview.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>5.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="5.1.4" data-path="enrichment-overview.html"><a href="enrichment-overview.html#other-gene-sets"><i class="fa fa-check"></i><b>5.1.4</b> Other gene sets</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="enrichment-overview.html"><a href="enrichment-overview.html#over-representation-analysis"><i class="fa fa-check"></i><b>5.2</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="5.3" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.3</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="enrichment-overview.html"><a href="enrichment-overview.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>5.4</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html"><i class="fa fa-check"></i><b>6</b> GO enrichment analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#supported-organisms"><i class="fa fa-check"></i><b>6.1</b> Supported organisms</a></li>
<li class="chapter" data-level="6.2" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-classification"><i class="fa fa-check"></i><b>6.2</b> GO classification</a></li>
<li class="chapter" data-level="6.3" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-over-representation-test"><i class="fa fa-check"></i><b>6.3</b> GO over-representation test</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>6.3.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="6.3.2" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>6.3.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="6.3.3" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>6.3.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.5" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-semantic-similarity-analysis"><i class="fa fa-check"></i><b>6.5</b> GO Semantic Similarity Analysis</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-analysis-for-non-model-organisms"><i class="fa fa-check"></i><b>6.5.1</b> GO analysis for non-model organisms</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html"><i class="fa fa-check"></i><b>7</b> KEGG enrichment analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#kegg-over-representation-test"><i class="fa fa-check"></i><b>7.1</b> KEGG over-representation test</a></li>
<li class="chapter" data-level="7.2" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#kegg-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>7.2</b> KEGG Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="7.3" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#kegg-module-over-representation-test"><i class="fa fa-check"></i><b>7.3</b> KEGG Module over-representation test</a></li>
<li class="chapter" data-level="7.4" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#kegg-module-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>7.4</b> KEGG Module Gene Set Enrichment Analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="reactomepa.html"><a href="reactomepa.html"><i class="fa fa-check"></i><b>8</b> Reactome enrichment analysis</a></li>
<li class="chapter" data-level="9" data-path="dose-enrichment.html"><a href="dose-enrichment.html"><i class="fa fa-check"></i><b>9</b> DO enrichment analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="dose-enrichment.html"><a href="dose-enrichment.html#enrichdo-function"><i class="fa fa-check"></i><b>9.1</b> <code>enrichDO</code> function</a></li>
<li class="chapter" data-level="9.2" data-path="dose-enrichment.html"><a href="dose-enrichment.html#enrichncg-function"><i class="fa fa-check"></i><b>9.2</b> <code>enrichNCG</code> function</a></li>
<li class="chapter" data-level="9.3" data-path="dose-enrichment.html"><a href="dose-enrichment.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>9.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
<li class="chapter" data-level="9.4" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsedo-fuction"><i class="fa fa-check"></i><b>9.4</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="9.5" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsencg-fuction"><i class="fa fa-check"></i><b>9.5</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="9.6" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsedgn-fuction"><i class="fa fa-check"></i><b>9.6</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="meshes-enrichment.html"><a href="meshes-enrichment.html"><i class="fa fa-check"></i><b>10</b> MeSH enrichment analysis</a></li>
<li class="chapter" data-level="11" data-path="universal-api.html"><a href="universal-api.html"><i class="fa fa-check"></i><b>11</b> Universal enrichment analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="universal-api.html"><a href="universal-api.html#input-data"><i class="fa fa-check"></i><b>11.1</b> Input data</a></li>
<li class="chapter" data-level="11.2" data-path="universal-api.html"><a href="universal-api.html#wikipathways-analysis"><i class="fa fa-check"></i><b>11.2</b> WikiPathways analysis</a></li>
<li class="chapter" data-level="11.3" data-path="universal-api.html"><a href="universal-api.html#cell-marker"><i class="fa fa-check"></i><b>11.3</b> Cell Marker</a></li>
<li class="chapter" data-level="11.4" data-path="universal-api.html"><a href="universal-api.html#msigdb-analysis"><i class="fa fa-check"></i><b>11.4</b> MSigDb analysis</a></li>
<li class="chapter" data-level="11.5" data-path="universal-api.html"><a href="universal-api.html#msigdb"><i class="fa fa-check"></i><b>11.5</b> MSigDb</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chipseeker-enrichment.html"><a href="chipseeker-enrichment.html"><i class="fa fa-check"></i><b>12</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="13" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html"><i class="fa fa-check"></i><b>13</b> Biological theme comparison</a>
<ul>
<li class="chapter" data-level="13.1" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>13.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="13.2" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>13.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="enrichplot.html"><a href="enrichplot.html"><i class="fa fa-check"></i><b>14</b> Visualization of functional enrichment result</a>
<ul>
<li class="chapter" data-level="14.1" data-path="enrichplot.html"><a href="enrichplot.html#bar-plot"><i class="fa fa-check"></i><b>14.1</b> Bar Plot</a></li>
<li class="chapter" data-level="14.2" data-path="enrichplot.html"><a href="enrichplot.html#dot-plot"><i class="fa fa-check"></i><b>14.2</b> Dot plot</a></li>
<li class="chapter" data-level="14.3" data-path="enrichplot.html"><a href="enrichplot.html#gene-concept-network"><i class="fa fa-check"></i><b>14.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="14.4" data-path="enrichplot.html"><a href="enrichplot.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>14.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="14.5" data-path="enrichplot.html"><a href="enrichplot.html#enrichment-map"><i class="fa fa-check"></i><b>14.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="14.6" data-path="enrichplot.html"><a href="enrichplot.html#upset-plot"><i class="fa fa-check"></i><b>14.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="14.7" data-path="enrichplot.html"><a href="enrichplot.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>14.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="14.8" data-path="enrichplot.html"><a href="enrichplot.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>14.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="14.9" data-path="enrichplot.html"><a href="enrichplot.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>14.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="14.10" data-path="enrichplot.html"><a href="enrichplot.html#goplot"><i class="fa fa-check"></i><b>14.10</b> goplot</a></li>
<li class="chapter" data-level="14.11" data-path="enrichplot.html"><a href="enrichplot.html#browsekegg"><i class="fa fa-check"></i><b>14.11</b> browseKEGG</a></li>
<li class="chapter" data-level="14.12" data-path="enrichplot.html"><a href="enrichplot.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>14.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="15" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html"><i class="fa fa-check"></i><b>15</b> dplyr verbs for manipulating enrichment result</a>
<ul>
<li class="chapter" data-level="15.1" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#filter"><i class="fa fa-check"></i><b>15.1</b> filter</a></li>
<li class="chapter" data-level="15.2" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#arrange"><i class="fa fa-check"></i><b>15.2</b> arrange</a></li>
<li class="chapter" data-level="15.3" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#select"><i class="fa fa-check"></i><b>15.3</b> select</a></li>
<li class="chapter" data-level="15.4" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#mutate"><i class="fa fa-check"></i><b>15.4</b> mutate</a></li>
<li class="chapter" data-level="15.5" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#slice"><i class="fa fa-check"></i><b>15.5</b> slice</a></li>
<li class="chapter" data-level="15.6" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#summarise"><i class="fa fa-check"></i><b>15.6</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="bitr.html"><a href="bitr.html"><i class="fa fa-check"></i><b>16</b> bitr: Biological Id TranslatoR</a>
<ul>
<li class="chapter" data-level="16.0.1" data-path="bitr.html"><a href="bitr.html#bitr_kegg"><i class="fa fa-check"></i><b>16.0.1</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="other-utilities.html"><a href="other-utilities.html"><i class="fa fa-check"></i><b>17</b> Useful utilities</a>
<ul>
<li class="chapter" data-level="17.1" data-path="other-utilities.html"><a href="other-utilities.html#setReadable"><i class="fa fa-check"></i><b>17.1</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="semantic-similarity-overview" class="section level1" number="1">
<h1><span class="header-section-number">Chapter 1</span> Overview of semantic similarity analysis</h1>
<p>Functional similarity of gene products can be estimated by controlled
biological vocabularies, such as Gene Ontology (GO), Disease Ontology (DO) and Medical Subject Headings (MeSH).</p>
<p>Four methods including Resnik <span class="citation">(Philip <a href="#ref-philip_semantic_1999" role="doc-biblioref">1999</a>)</span>, Jiang <span class="citation">(Jiang and Conrath <a href="#ref-jiang_semantic_1997" role="doc-biblioref">1997</a>)</span>, Lin <span class="citation">(Lin <a href="#ref-lin_information-theoretic_1998" role="doc-biblioref">1998</a>)</span> and Schlicker <span class="citation">(Schlicker et al. <a href="#ref-schlicker_new_2006" role="doc-biblioref">2006</a>)</span> have been presented to determine the semantic similarity of two GO terms based on the annotation statistics of their common ancestor terms. Wang <span class="citation">(Wang et al. <a href="#ref-wang_new_2007" role="doc-biblioref">2007</a>)</span>
proposed a method to measure the similarity based on the graph structure of GO. Each of these methods has its own advantages and
weaknesses and can be applied to other ontologies that have similar structure (i.e.¬†directed acyclic graph).</p>
<div id="information-content-based-methods" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Information content-based methods</h2>
<p>Four methods proposed by Resnik <span class="citation">(Philip <a href="#ref-philip_semantic_1999" role="doc-biblioref">1999</a>)</span>,
Jiang <span class="citation">(Jiang and Conrath <a href="#ref-jiang_semantic_1997" role="doc-biblioref">1997</a>)</span>, Lin <span class="citation">(Lin <a href="#ref-lin_information-theoretic_1998" role="doc-biblioref">1998</a>)</span>
and Schlicker <span class="citation">(Schlicker et al. <a href="#ref-schlicker_new_2006" role="doc-biblioref">2006</a>)</span> are information content (IC) based, which depend on the frequencies of two GO terms involved and that of their closest common ancestor term in a specific corpus of GO
annotations. The information content of a GO term is computed by the
negative log probability of the term occurring in GO corpus. A rarely used term contains a greater amount of information.</p>
<p>The frequency of a term t is defined as:</p>
<p><span class="math display">\[p(t) = \frac{n_{t&#39;}}{N} | t&#39; \in \left\{t, \; children\: of\: t \right\}\]</span></p>
<p>where <span class="math inline">\(n_{t&#39;}\)</span> is the number of term <span class="math inline">\(t&#39;\)</span>, and <span class="math inline">\(N\)</span> is the total number of terms in GO corpus.</p>
<p>Thus the information content is defined as:</p>
<p><span class="math display">\[IC(t) = -\log(p(t))\]</span></p>
<p>As GO allow multiple parents for each concept, two terms can share
parents by multiple paths. IC-based methods calculate similarity of two GO terms based on the information content of their closest common ancestor term, which was also called most informative common ancestor (MICA).</p>
<div id="resnik-method" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Resnik method</h3>
<p>The Resnik method is defined as:</p>
<p><span class="math display">\[sim_{Resnik}(t_1,t_2) = IC(MICA)\]</span></p>
</div>
<div id="lin-method" class="section level3" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> Lin method</h3>
<p>The Lin method is defined as:</p>
<p><span class="math display">\[sim_{Lin}(t_1,t_2) = \frac{2IC(MICA)}{IC(t_1)+IC(t_2)}\]</span></p>
</div>
<div id="rel-method" class="section level3" number="1.1.3">
<h3><span class="header-section-number">1.1.3</span> Rel method</h3>
<p>The Relevance method, which was proposed by Schlicker, combine Resnik‚Äôs and Lin‚Äôs method and is defined as:</p>
<p><span class="math display">\[sim_{Rel}(t_1,t_2) = \frac{2IC(MICA)(1-p(MICA))}{IC(t_1)+IC(t_2)}\]</span></p>
</div>
<div id="jiang-method" class="section level3" number="1.1.4">
<h3><span class="header-section-number">1.1.4</span> Jiang method</h3>
<p>The Jiang and Conrath‚Äôs method is defined as:</p>
<p><span class="math display">\[sim_{Jiang}(t_1,t_2) = 1-\min(1, IC(t_1) + IC(t_2) - 2IC(MICA))\]</span></p>
</div>
</div>
<div id="graph-based-method" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Graph-based method</h2>
<p>Graph-based methods using the topology of GO graph structure to
compute semantic similarity. Formally, a GO term A can be represented
as <span class="math inline">\(DAG_{A}=(A,T_{A},E_{A})\)</span> where <span class="math inline">\(T_{A}\)</span> is the set of GO terms in
<span class="math inline">\(DAG_{A}\)</span>, including term A and all of its ancestor terms in the GO
graph, and <span class="math inline">\(E_{A}\)</span> is the set of edges connecting the GO terms in
<span class="math inline">\(DAG_{A}\)</span>.</p>
<div id="wang-method" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Wang method</h3>
<p>To encode the semantic of a GO term in a measurable format to enable a quantitative comparison, Wang<span class="citation">(Wang et al. <a href="#ref-wang_new_2007" role="doc-biblioref">2007</a>)</span> firstly defined the semantic value of term A as the aggregate contribution of all terms in <span class="math inline">\(DAG_{A}\)</span> to the semantics of term A, terms closer to term A in <span class="math inline">\(DAG_{A}\)</span> contribute more to its semantics. Thus, defined the contribution of a GO term <span class="math inline">\(t\)</span> to the semantic of GO term <span class="math inline">\(A\)</span> as the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(A\)</span>.</p>
<p>For any of term <span class="math inline">\(t\)</span> in <span class="math inline">\(DAG_{A}\)</span>, its S-value related to term <span class="math inline">\(A\)</span>, <span class="math inline">\(S_{A}(\textit{t})\)</span> is defined as:</p>
<p><span class="math display">\[\left\{\begin{array}{l} S_{A}(A)=1 \\ S_{A}(\textit{t})=\max\{w_{e} \times S_{A}(\textit{t}&#39;) | \textit{t}&#39; \in children \: of(\textit{t}) \} \; if \: \textit{t} \ne A \end{array} \right.\]</span></p>
<p>where <span class="math inline">\(w_{e}\)</span> is the semantic contribution factor for edge <span class="math inline">\(e \in E_{A}\)</span> linking term <span class="math inline">\(t\)</span> with its child term <span class="math inline">\(t&#39;\)</span>.
Term <span class="math inline">\(A\)</span> contributes to its own is defined as 1. After obtaining the S-values for all terms in <span class="math inline">\(DAG_{A}\)</span>,
the semantic value of DO term A, <span class="math inline">\(SV(A)\)</span>, is calculated as:</p>
<p><span class="math display">\[SV(A)=\displaystyle\sum_{t \in T_{A}} S_{A}(t)\]</span></p>
<p>Thus given two GO terms A and B, the semantic similarity between these two terms is defined as:</p>
<p><span class="math display">\[sim_{Wang}(A, B) = \frac{\displaystyle\sum_{t \in T_{A} \cap T_{B}}{S_{A}(t) + S_{B}(t)}}{SV(A) + SV(B)}\]</span></p>
<p>where <span class="math inline">\(S_{A}(\textit{t})\)</span> is the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(A\)</span>
and <span class="math inline">\(S_{B}(\textit{t})\)</span> is the S-value of GO term <span class="math inline">\(t\)</span> related to term <span class="math inline">\(B\)</span>.</p>
<p>This method proposed by Wang <span class="citation">(Wang et al. <a href="#ref-wang_new_2007" role="doc-biblioref">2007</a>)</span> determines the semantic
similarity of two GO terms based on both the locations of these terms
in the GO graph and their relations with their ancestor terms.</p>
</div>
</div>
<div id="combine-methods" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Combine methods</h2>
<p>Since a gene product can be annotated by multiple GO terms, semantic similarity among gene products needs to be aggregated from different semantic similarity scores of multiple GO terms associated with genes, including <code>max</code>, <code>avg</code>, <code>rcmax</code> and <code>BMA</code>.</p>
<div id="max" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> max</h3>
<p>The <code>max</code> method calculates the maximum semantic similarity score over all pairs of GO terms between these two GO term sets.</p>
<p><span class="math display">\[sim_{max}(g_1, g_2) = \displaystyle\max_{1 \le i \le m, 1 \le j \le n} sim(go_{1i}, go_{2j})\]</span></p>
</div>
<div id="avg" class="section level3" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> avg</h3>
<p>The <code>avg</code> calculates the average semantic similarity score over all pairs of GO terms.</p>
<p><span class="math display">\[sim_{avg}(g_1, g_2) = \frac{\displaystyle\sum_{i=1}^m\sum_{j=1}^nsim(go_{1i}, go_{2j})}{m \times n}\]</span></p>
</div>
<div id="rcmax" class="section level3" number="1.3.3">
<h3><span class="header-section-number">1.3.3</span> rcmax</h3>
<p>Similarities among two sets of GO terms form a matrix, the <code>rcmax</code> method uses the maximum of <code>RowScore</code> and <code>ColumnScore</code>, where <code>RowScore</code> (or <code>ColumnScore</code>) is the average of maximum similarity on each row (or column).</p>
<p><span class="math display">\[sim_{rcmax}(g_1, g_2) = \max(\frac{\displaystyle\sum_{i=1}^m \max_{1 \le j \le n} sim(go_{1i}, go_{2j})}{m},\frac{\displaystyle\sum_{j=1}^n \max_{1 \le i \le m} sim(go_{1i},go_{2j})}{n})\]</span></p>
</div>
<div id="bma" class="section level3" number="1.3.4">
<h3><span class="header-section-number">1.3.4</span> BMA</h3>
<p>The <code>BMA</code> method, used the <strong>B</strong>est-<strong>M</strong>atch <strong>A</strong>verage strategy, calculates the average of all maximum similarities on each row and column, and is defined as:</p>
<p><span class="math display">\[sim_{BMA}(g_1, g_2) = \frac{\displaystyle\sum_{1=i}^m \max_{1 \le j \le n}sim(go_{1i}, go_{2j}) + \displaystyle\sum_{1=j}^n \max_{1 \le i \le m}sim(go_{1i}, go_{2j})} {m+n}\]</span></p>
</div>
</div>
<div id="summary" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Summary</h2>
<p>The idea behind semantic similarity measurement is the notion that genes with similar function should have similar annotation vocabulary and have a close relationship in the ontology strucutre. Measuring similarity is critical for expanding knownledge, since similar objects tend to behave similarly, which supports many bioinformatics applications to infer gene/protein functions, miRNA function, genetic interaction, protein-protein interaction, miRNA-mRNA interaction and celluar localization.</p>
<p>We developed several Bioconductor packages, including <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> <span class="citation">(Yu et al. <a href="#ref-yu2010" role="doc-biblioref">2010</a>; Yu <a href="#ref-yu_gosemsim_2020" role="doc-biblioref">2020</a>)</span> for computing semantic similarity among GO terms, sets of GO terms, gene products and gene clusters (see also <a href="GOSemSim.html#GOSemSim">Chapter 2</a>), <a href="http://bioconductor.org/packages/DOSE">DOSE</a> <span class="citation">(G. Yu et al. <a href="#ref-yu_dose_2015" role="doc-biblioref">2015</a>)</span> for Disease Ontology (DO) (see also <a href="DOSE-semantic-similarity.html#DOSE-semantic-similarity">Chapter 3</a>) and <a href="http://bioconductor.org/packages/meshes">meshes</a> <span class="citation">(Yu <a href="#ref-yu_meshes_2018" role="doc-biblioref">2018</a>)</span> that based on Medical Subject Headings (MeSH) (see also <a href="meshes-semantic-similarity.html#meshes-semantic-similarity">Chapter 4</a>).</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-jiang_semantic_1997">
<p>Jiang, Jay J., and David W. Conrath. 1997. ‚ÄúSemantic Similarity Based on Corpus Statistics and Lexical Taxonomy.‚Äù <em>Proceedings of 10th International Conference on Research in Computational Linguistics</em>. <a href="http://www.citebase.org/abstract?id=oai:arXiv.org:cmp-lg/9709008">http://www.citebase.org/abstract?id=oai:arXiv.org:cmp-lg/9709008</a>.</p>
</div>
<div id="ref-lin_information-theoretic_1998">
<p>Lin, Dekang. 1998. ‚ÄúAn Information-Theoretic Definition of Similarity.‚Äù <em>In Proceedings of the 15th International Conference on Machine Learning</em>, 296‚Äî304. <a href="https://doi.org/10.1.1.55.1832">https://doi.org/10.1.1.55.1832</a>.</p>
</div>
<div id="ref-philip_semantic_1999">
<p>Philip, Resnik. 1999. ‚ÄúSemantic Similarity in a Taxonomy: An Information-Based Measure and Its Application to Problems of Ambiguity in Natural Language.‚Äù <em>Journal of Artificial Intelligence Research</em> 11: 95‚Äì130.</p>
</div>
<div id="ref-schlicker_new_2006">
<p>Schlicker, Andreas, Francisco S Domingues, J√∂rg Rahnenf√ºhrer, and Thomas Lengauer. 2006. ‚ÄúA New Measure for Functional Similarity of Gene Products Based on Gene Ontology.‚Äù <em>BMC Bioinformatics</em> 7: 302. <a href="https://doi.org/1471-2105-7-302">https://doi.org/1471-2105-7-302</a>.</p>
</div>
<div id="ref-wang_new_2007">
<p>Wang, James Z, Zhidian Du, Rapeeporn Payattakool, Philip S Yu, and Chin-Fu Chen. 2007. ‚ÄúA New Method to Measure the Semantic Similarity of Go Terms.‚Äù <em>Bioinformatics (Oxford, England)</em> 23 (May): 1274‚Äì81. <a href="https://doi.org/btm087">https://doi.org/btm087</a>.</p>
</div>
<div id="ref-yu_gosemsim_2020">
<p>Yu, Guangchuang. 2020. ‚ÄúGene Ontology Semantic Similarity Analysis Using GOSemSim.‚Äù <em>Methods in Molecular Biology (Clifton, N.J.)</em> 2117: 207‚Äì15. <a href="https://doi.org/10.1007/978-1-0716-0301-7_11">https://doi.org/10.1007/978-1-0716-0301-7_11</a>.</p>
</div>
<div id="ref-yu_meshes_2018">
<p>Yu, Guangchuang. 2018. ‚ÄúUsing Meshes for MeSH Term Enrichment and Semantic Analyses.‚Äù <em>Bioinformatics</em> 34 (21): 3766‚Äì7. <a href="https://doi.org/10.1093/bioinformatics/bty410">https://doi.org/10.1093/bioinformatics/bty410</a>.</p>
</div>
<div id="ref-yu2010">
<p>Yu, Guangchuang, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu, and Shengqi Wang. 2010. ‚ÄúGOSemSim: An R Package for Measuring Semantic Similarity Among Go Terms and Gene Products.‚Äù <em>Bioinformatics</em> 26: 976‚Äì78. <a href="https://doi.org/10.1093/bioinformatics/btq064">https://doi.org/10.1093/bioinformatics/btq064</a>.</p>
</div>
<div id="ref-yu_dose_2015">
<p>Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. ‚ÄúDOSE: An R/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.‚Äù <em>Bioinformatics</em> 31 (4): 608‚Äì9. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="GOSemSim.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
