<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Universal enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler</title>
  <meta name="description" content="Chapter 12 Universal enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Universal enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Universal enrichment analysis | Biomedical Knowledge Mining using GOSemSim and clusterProfiler" />
  
  
  



<meta name="date" content="2020-09-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="meshes-enrichment.html"/>
<link rel="next" href="chipseeker-enrichment.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.15/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>üìñ Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>üéØ Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i>üìù Citation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-structure"><i class="fa fa-check"></i>üìö Book structure</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#want-to-help"><i class="fa fa-check"></i>üíñ Want to help?</a></li>
</ul></li>
<li class="part"><span><b>Part I: Semantic Similarity Analysis</b></span></li>
<li class="chapter" data-level="1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html"><i class="fa fa-check"></i><b>1</b> Overview of semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="1.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#information-content-based-methods"><i class="fa fa-check"></i><b>1.1</b> Information content-based methods</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#resnik-method"><i class="fa fa-check"></i><b>1.1.1</b> Resnik method</a></li>
<li class="chapter" data-level="1.1.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#lin-method"><i class="fa fa-check"></i><b>1.1.2</b> Lin method</a></li>
<li class="chapter" data-level="1.1.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#rel-method"><i class="fa fa-check"></i><b>1.1.3</b> Rel method</a></li>
<li class="chapter" data-level="1.1.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#jiang-method"><i class="fa fa-check"></i><b>1.1.4</b> Jiang method</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#graph-based-method"><i class="fa fa-check"></i><b>1.2</b> Graph-based method</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#wang-method"><i class="fa fa-check"></i><b>1.2.1</b> Wang method</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#combine-methods"><i class="fa fa-check"></i><b>1.3</b> Combine methods</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#max"><i class="fa fa-check"></i><b>1.3.1</b> max</a></li>
<li class="chapter" data-level="1.3.2" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#avg"><i class="fa fa-check"></i><b>1.3.2</b> avg</a></li>
<li class="chapter" data-level="1.3.3" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#rcmax"><i class="fa fa-check"></i><b>1.3.3</b> rcmax</a></li>
<li class="chapter" data-level="1.3.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#bma"><i class="fa fa-check"></i><b>1.3.4</b> BMA</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="semantic-similarity-overview.html"><a href="semantic-similarity-overview.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="GOSemSim.html"><a href="GOSemSim.html"><i class="fa fa-check"></i><b>2</b> GO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="GOSemSim.html"><a href="GOSemSim.html#semantic-data"><i class="fa fa-check"></i><b>2.1</b> Semantic data</a></li>
<li class="chapter" data-level="2.2" data-path="GOSemSim.html"><a href="GOSemSim.html#gosemsim-supported-organisms"><i class="fa fa-check"></i><b>2.2</b> Supported organisms</a></li>
<li class="chapter" data-level="2.3" data-path="GOSemSim.html"><a href="GOSemSim.html#go-semantic-simiarlity"><i class="fa fa-check"></i><b>2.3</b> GO semantic similarity measurement</a></li>
<li class="chapter" data-level="2.4" data-path="GOSemSim.html"><a href="GOSemSim.html#gene-go-semantic-similarity"><i class="fa fa-check"></i><b>2.4</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="2.5" data-path="GOSemSim.html"><a href="GOSemSim.html#gene-cluster-go-semantic-similarity"><i class="fa fa-check"></i><b>2.5</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html"><i class="fa fa-check"></i><b>3</b> DO semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#do-semantic-similarity"><i class="fa fa-check"></i><b>3.1</b> DO term semantic similarity measurement</a></li>
<li class="chapter" data-level="3.2" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#gene-do-semantic-similarity"><i class="fa fa-check"></i><b>3.2</b> Gene semantic similarity measurement</a></li>
<li class="chapter" data-level="3.3" data-path="DOSE-semantic-similarity.html"><a href="DOSE-semantic-similarity.html#gene-cluster-do-semantic-similarity"><i class="fa fa-check"></i><b>3.3</b> Gene cluster semantic similarity measurement</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html"><i class="fa fa-check"></i><b>4</b> MeSH semantic similarity analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#meshes-supported-organisms"><i class="fa fa-check"></i><b>4.1</b> Supported organisms</a></li>
<li class="chapter" data-level="4.2" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#meshes-semantic-simiarlity"><i class="fa fa-check"></i><b>4.2</b> MeSH semantic similarity measurement</a></li>
<li class="chapter" data-level="4.3" data-path="meshes-semantic-similarity.html"><a href="meshes-semantic-similarity.html#gene-meshes-semantic-similarity"><i class="fa fa-check"></i><b>4.3</b> Gene semantic similarity measurement</a></li>
</ul></li>
<li class="part"><span><b>Part II: Enrichment analysis</b></span></li>
<li class="chapter" data-level="5" data-path="enrichment-overview.html"><a href="enrichment-overview.html"><i class="fa fa-check"></i><b>5</b> Overview of enrichment analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="enrichment-overview.html"><a href="enrichment-overview.html#terminology"><i class="fa fa-check"></i><b>5.1</b> Terminology</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>5.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="5.1.2" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gene-ontology-go"><i class="fa fa-check"></i><b>5.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="5.1.3" data-path="enrichment-overview.html"><a href="enrichment-overview.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>5.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="5.1.4" data-path="enrichment-overview.html"><a href="enrichment-overview.html#other-gene-sets"><i class="fa fa-check"></i><b>5.1.4</b> Other gene sets</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="enrichment-overview.html"><a href="enrichment-overview.html#ora-algorithm"><i class="fa fa-check"></i><b>5.2</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="5.3" data-path="enrichment-overview.html"><a href="enrichment-overview.html#gsea-algorithm"><i class="fa fa-check"></i><b>5.3</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="enrichment-overview.html"><a href="enrichment-overview.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>5.4</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html"><i class="fa fa-check"></i><b>6</b> GO enrichment analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterProfiler-go-supported-organisms"><i class="fa fa-check"></i><b>6.1</b> Supported organisms</a></li>
<li class="chapter" data-level="6.2" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-classification"><i class="fa fa-check"></i><b>6.2</b> GO classification</a></li>
<li class="chapter" data-level="6.3" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-ora"><i class="fa fa-check"></i><b>6.3</b> GO over-representation analysis</a></li>
<li class="chapter" data-level="6.4" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.5" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-non-model"><i class="fa fa-check"></i><b>6.5</b> GO analysis for non-model organisms</a></li>
<li class="chapter" data-level="6.6" data-path="clusterprofiler-go.html"><a href="clusterprofiler-go.html#clusterprofiler-go-summary"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html"><i class="fa fa-check"></i><b>7</b> KEGG enrichment analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterProfiler-kegg-supported-organisms"><i class="fa fa-check"></i><b>7.1</b> Supported organisms</a></li>
<li class="chapter" data-level="7.2" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-pathway-ora"><i class="fa fa-check"></i><b>7.2</b> KEGG pathway over-representation analysis</a></li>
<li class="chapter" data-level="7.3" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-pathway-gsea"><i class="fa fa-check"></i><b>7.3</b> KEGG pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="7.4" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-module-ora"><i class="fa fa-check"></i><b>7.4</b> KEGG module over-representation analysis</a></li>
<li class="chapter" data-level="7.5" data-path="clusterprofiler-kegg.html"><a href="clusterprofiler-kegg.html#clusterprofiler-kegg-module-gsea"><i class="fa fa-check"></i><b>7.5</b> KEGG module gene set enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="wikipathways-analysis.html"><a href="wikipathways-analysis.html"><i class="fa fa-check"></i><b>8</b> WikiPathways analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="wikipathways-analysis.html"><a href="wikipathways-analysis.html#clusterprofiler-wikipathway-ora"><i class="fa fa-check"></i><b>8.1</b> WikiPathways over-representation analysis</a></li>
<li class="chapter" data-level="8.2" data-path="wikipathways-analysis.html"><a href="wikipathways-analysis.html#clusterprofiler-wikipathway-gsea"><i class="fa fa-check"></i><b>8.2</b> WikiPathways gene set enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="reactomepa.html"><a href="reactomepa.html"><i class="fa fa-check"></i><b>9</b> Reactome enrichment analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-supported-organisms"><i class="fa fa-check"></i><b>9.1</b> Supported organisms</a></li>
<li class="chapter" data-level="9.2" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-ora"><i class="fa fa-check"></i><b>9.2</b> Reactome pathway over-representation analysis</a></li>
<li class="chapter" data-level="9.3" data-path="reactomepa.html"><a href="reactomepa.html#reactomepa-gsea"><i class="fa fa-check"></i><b>9.3</b> Reactome pathway gene set enrichment analysis</a></li>
<li class="chapter" data-level="9.4" data-path="reactomepa.html"><a href="reactomepa.html#pathway-visualization"><i class="fa fa-check"></i><b>9.4</b> Pathway Visualization</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="dose-enrichment.html"><a href="dose-enrichment.html"><i class="fa fa-check"></i><b>10</b> Disease enrichment analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="dose-enrichment.html"><a href="dose-enrichment.html#dose-ora"><i class="fa fa-check"></i><b>10.1</b> Disease over-representation analysis</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="dose-enrichment.html"><a href="dose-enrichment.html#dose-do-ora"><i class="fa fa-check"></i><b>10.1.1</b> Over-representation analysis for disease ontology</a></li>
<li class="chapter" data-level="10.1.2" data-path="dose-enrichment.html"><a href="dose-enrichment.html#dose-ncg-ora"><i class="fa fa-check"></i><b>10.1.2</b> Over-representation analysis for the network of cancer gene</a></li>
<li class="chapter" data-level="10.1.3" data-path="dose-enrichment.html"><a href="dose-enrichment.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>10.1.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="dose-enrichment.html"><a href="dose-enrichment.html#dose-gsea"><i class="fa fa-check"></i><b>10.2</b> Disease gene set enrichment analysis</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsedo-fuction"><i class="fa fa-check"></i><b>10.2.1</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="10.2.2" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsencg-fuction"><i class="fa fa-check"></i><b>10.2.2</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="10.2.3" data-path="dose-enrichment.html"><a href="dose-enrichment.html#gsedgn-fuction"><i class="fa fa-check"></i><b>10.2.3</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="meshes-enrichment.html"><a href="meshes-enrichment.html"><i class="fa fa-check"></i><b>11</b> MeSH enrichment analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="meshes-enrichment.html"><a href="meshes-enrichment.html#meshes-ora"><i class="fa fa-check"></i><b>11.1</b> MeSH over-representation analysis</a></li>
<li class="chapter" data-level="11.2" data-path="meshes-enrichment.html"><a href="meshes-enrichment.html#meshes-gsea"><i class="fa fa-check"></i><b>11.2</b> MeSH gene set enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="universal-api.html"><a href="universal-api.html"><i class="fa fa-check"></i><b>12</b> Universal enrichment analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="universal-api.html"><a href="universal-api.html#input-data"><i class="fa fa-check"></i><b>12.1</b> Input data</a></li>
<li class="chapter" data-level="12.2" data-path="universal-api.html"><a href="universal-api.html#cell-marker"><i class="fa fa-check"></i><b>12.2</b> Cell Marker</a></li>
<li class="chapter" data-level="12.3" data-path="universal-api.html"><a href="universal-api.html#msigdb-analysis"><i class="fa fa-check"></i><b>12.3</b> MSigDb analysis</a></li>
<li class="chapter" data-level="12.4" data-path="universal-api.html"><a href="universal-api.html#msigdb"><i class="fa fa-check"></i><b>12.4</b> MSigDb</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chipseeker-enrichment.html"><a href="chipseeker-enrichment.html"><i class="fa fa-check"></i><b>13</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="14" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html"><i class="fa fa-check"></i><b>14</b> Biological theme comparison</a>
<ul>
<li class="chapter" data-level="14.1" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>14.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="14.2" data-path="clusterprofiler-comparecluster.html"><a href="clusterprofiler-comparecluster.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>14.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="enrichplot.html"><a href="enrichplot.html"><i class="fa fa-check"></i><b>15</b> Visualization of functional enrichment result</a>
<ul>
<li class="chapter" data-level="15.1" data-path="enrichplot.html"><a href="enrichplot.html#bar-plot"><i class="fa fa-check"></i><b>15.1</b> Bar Plot</a></li>
<li class="chapter" data-level="15.2" data-path="enrichplot.html"><a href="enrichplot.html#dot-plot"><i class="fa fa-check"></i><b>15.2</b> Dot plot</a></li>
<li class="chapter" data-level="15.3" data-path="enrichplot.html"><a href="enrichplot.html#gene-concept-network"><i class="fa fa-check"></i><b>15.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="15.4" data-path="enrichplot.html"><a href="enrichplot.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>15.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="15.5" data-path="enrichplot.html"><a href="enrichplot.html#enrichment-map"><i class="fa fa-check"></i><b>15.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="15.6" data-path="enrichplot.html"><a href="enrichplot.html#upset-plot"><i class="fa fa-check"></i><b>15.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="15.7" data-path="enrichplot.html"><a href="enrichplot.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>15.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="15.8" data-path="enrichplot.html"><a href="enrichplot.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>15.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="15.9" data-path="enrichplot.html"><a href="enrichplot.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>15.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="15.10" data-path="enrichplot.html"><a href="enrichplot.html#goplot"><i class="fa fa-check"></i><b>15.10</b> goplot</a></li>
<li class="chapter" data-level="15.11" data-path="enrichplot.html"><a href="enrichplot.html#browsekegg"><i class="fa fa-check"></i><b>15.11</b> browseKEGG</a></li>
<li class="chapter" data-level="15.12" data-path="enrichplot.html"><a href="enrichplot.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>15.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="16" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html"><i class="fa fa-check"></i><b>16</b> dplyr verbs for manipulating enrichment result</a>
<ul>
<li class="chapter" data-level="16.1" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#filter"><i class="fa fa-check"></i><b>16.1</b> filter</a></li>
<li class="chapter" data-level="16.2" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#arrange"><i class="fa fa-check"></i><b>16.2</b> arrange</a></li>
<li class="chapter" data-level="16.3" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#select"><i class="fa fa-check"></i><b>16.3</b> select</a></li>
<li class="chapter" data-level="16.4" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#mutate"><i class="fa fa-check"></i><b>16.4</b> mutate</a></li>
<li class="chapter" data-level="16.5" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#slice"><i class="fa fa-check"></i><b>16.5</b> slice</a></li>
<li class="chapter" data-level="16.6" data-path="clusterProfiler-dplyr.html"><a href="clusterProfiler-dplyr.html#summarise"><i class="fa fa-check"></i><b>16.6</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="useful-utilities.html"><a href="useful-utilities.html"><i class="fa fa-check"></i><b>17</b> Useful utilities</a>
<ul>
<li class="chapter" data-level="17.1" data-path="useful-utilities.html"><a href="useful-utilities.html#id-convert"><i class="fa fa-check"></i><b>17.1</b> Translating Biological ID</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="useful-utilities.html"><a href="useful-utilities.html#bitr"><i class="fa fa-check"></i><b>17.1.1</b> bitr: Biological Id TranslatoR</a></li>
<li class="chapter" data-level="17.1.2" data-path="useful-utilities.html"><a href="useful-utilities.html#bitr_kegg"><i class="fa fa-check"></i><b>17.1.2</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="useful-utilities.html"><a href="useful-utilities.html#setReadable"><i class="fa fa-check"></i><b>17.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
<li class="chapter" data-level="17.3" data-path="useful-utilities.html"><a href="useful-utilities.html#go-utilities"><i class="fa fa-check"></i><b>17.3</b> GO utilities</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="useful-utilities.html"><a href="useful-utilities.html#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>17.3.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="17.3.2" data-path="useful-utilities.html"><a href="useful-utilities.html#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>17.3.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="17.3.3" data-path="useful-utilities.html"><a href="useful-utilities.html#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>17.3.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a>
<ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#genelist"><i class="fa fa-check"></i><b>A.1</b> How to prepare your own geneList</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#no-gene-can-be-mapped"><i class="fa fa-check"></i><b>A.2</b> No gene can be mapped</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Knowledge Mining using GOSemSim and clusterProfiler</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="universal-api" class="section level1" number="12">
<h1><span class="header-section-number">Chapter 12</span> Universal enrichment analysis</h1>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package <span class="citation">(Yu et al. <a href="#ref-yu2012" role="doc-biblioref">2012</a>)</span> supports both hypergeometric test and gene set enrichment analyses of many ontology/pathway, but it‚Äôs still not enough for users may want to analyze their data with unsupported organisms, slim version of GO, novel functional annotation (e.g.¬†GO via BlastGO or KEGG via KAAS), unsupported ontologies/pathways or customized annotations.</p>
<p>The <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> package provides <code>enricher()</code> function for hypergeometric test and <code>GSEA()</code> function for gene set enrichment analysis that are designed to accept user defined annotation. They accept two additional parameters <code>TERM2GENE</code> and <code>TERM2NAME</code>. As indicated in the parameter names, <code>TERM2GENE</code> is a data.frame with first column of term ID and second column of corresponding mapped gene and <code>TERM2NAME</code> is a <code>data.frame</code> with first column of term ID and second column of corresponding term name. <code>TERM2NAME</code> is optional.</p>
<div id="input-data" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Input data</h2>
<p>For over representation analysis, all we need is a gene vector, that is a vector of gene IDs. These gene IDs can be obtained by differential expression analysis (<em>e.g.</em> with the <a href="http://bioconductor.org/packages/DESeq2">DESeq2</a> package).</p>
<p>For gene set enrichment analysis, we need a ranked list of genes. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides an example dataset <code>geneList</code> which was derived from <code>R</code> package <a href="http://bioconductor.org/packages/breastCancerMAINZ">breastCancerMAINZ</a> that contained 200 samples, including 29 samples in grade I, 136 samples in grade II and 35 samples in grade III. We computed the ratios of geometric means of grade III samples versus geometric means of grade I samples. Logarithm of these ratios (base 2) were stored in <code>geneList</code> dataset. If you want to prepare your own <code>geneList</code>, please refer to the <a href="faq.html#genelist">FAQ</a>.</p>
<p>We can load the sample data into R via:</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="universal-api.html#cb115-1" aria-hidden="true"></a><span class="kw">data</span>(geneList, <span class="dt">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb115-2"><a href="universal-api.html#cb115-2" aria-hidden="true"></a><span class="kw">head</span>(geneList)</span></code></pre></div>
<pre><code>##     4312     8318    10874    55143    55388      991 
## 4.572613 4.514594 4.418218 4.144075 3.876258 3.677857</code></pre>
<p>Suppose we define fold change greater than 2 as DEGs:</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="universal-api.html#cb117-1" aria-hidden="true"></a>gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]</span>
<span id="cb117-2"><a href="universal-api.html#cb117-2" aria-hidden="true"></a><span class="kw">head</span>(gene)</span></code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
</div>
<div id="cell-marker" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Cell Marker</h2>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="universal-api.html#cb119-1" aria-hidden="true"></a>cell_markers &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(<span class="st">&#39;http://bio-bigdata.hrbmu.edu.cn/CellMarker/download/Human_cell_markers.txt&#39;</span>) <span class="op">%&gt;%</span></span>
<span id="cb119-2"><a href="universal-api.html#cb119-2" aria-hidden="true"></a><span class="st">   </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;cellMarker&quot;</span>, tissueType, cancerType, cellName, <span class="dt">sep=</span><span class="st">&quot;, &quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb119-3"><a href="universal-api.html#cb119-3" aria-hidden="true"></a><span class="st">   </span>dplyr<span class="op">::</span><span class="kw">select</span>(cellMarker, geneID) <span class="op">%&gt;%</span></span>
<span id="cb119-4"><a href="universal-api.html#cb119-4" aria-hidden="true"></a><span class="st">   </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">geneID =</span> <span class="kw">strsplit</span>(geneID, <span class="st">&#39;, &#39;</span>))</span>
<span id="cb119-5"><a href="universal-api.html#cb119-5" aria-hidden="true"></a>cell_markers</span></code></pre></div>
<pre><code>## # A tibble: 2,868 x 2
##    cellMarker                                             geneID   
##    &lt;chr&gt;                                                  &lt;list&gt;   
##  1 Kidney, Normal, Proximal tubular cell                  &lt;chr [1]&gt;
##  2 Liver, Normal, Ito cell (hepatic stellate cell)        &lt;chr [1]&gt;
##  3 Endometrium, Normal, Trophoblast cell                  &lt;chr [1]&gt;
##  4 Germ, Normal, Primordial germ cell                     &lt;chr [1]&gt;
##  5 Corneal epithelium, Normal, Epithelial cell            &lt;chr [1]&gt;
##  6 Placenta, Normal, Cytotrophoblast                      &lt;chr [1]&gt;
##  7 Periosteum, Normal, Periosteum-derived progenitor cell &lt;chr [4]&gt;
##  8 Amniotic membrane, Normal, Amnion epithelial cell      &lt;chr [2]&gt;
##  9 Primitive streak, Normal, Primitive streak cell        &lt;chr [2]&gt;
## 10 Adipose tissue, Normal, Stromal vascular fraction cell &lt;chr [1]&gt;
## # ‚Ä¶ with 2,858 more rows</code></pre>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="universal-api.html#cb121-1" aria-hidden="true"></a>y &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE=</span>cell_markers, <span class="dt">minGSSize=</span><span class="dv">1</span>)</span>
<span id="cb121-2"><a href="universal-api.html#cb121-2" aria-hidden="true"></a>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="kw">as.data.frame</span>(y))</span></code></pre></div>
<div id="htmlwidget-5b0083ca349889ed1d78" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5b0083ca349889ed1d78">{"x":{"filter":"none","data":[["Embryonic prefrontal cortex, Normal, Neural progenitor cell","Large intestine, Normal, MKI67+ progenitor cell","Kidney, Normal, Neutrophil","Large intestine, Normal, DCLK1+ progenitor cell","Kidney, Renal Cell Carcinoma, Neutrophil","Fetal gonad, Normal, Endothelial cell"],["Embryonic prefrontal cortex, Normal, Neural progenitor cell","Large intestine, Normal, MKI67+ progenitor cell","Kidney, Normal, Neutrophil","Large intestine, Normal, DCLK1+ progenitor cell","Kidney, Renal Cell Carcinoma, Neutrophil","Fetal gonad, Normal, Endothelial cell"],["Embryonic prefrontal cortex, Normal, Neural progenitor cell","Large intestine, Normal, MKI67+ progenitor cell","Kidney, Normal, Neutrophil","Large intestine, Normal, DCLK1+ progenitor cell","Kidney, Renal Cell Carcinoma, Neutrophil","Fetal gonad, Normal, Endothelial cell"],["33/153","20/153","6/153","7/153","6/153","9/153"],["166/11582","111/11582","42/11582","99/11582","77/11582","209/11582"],[4.43518066168319e-30,1.42606245514518e-17,1.70645935795602e-05,0.000329875455341899,0.0005276542867749,0.00181179940206466],[6.34230834620697e-28,1.0196346554288e-15,0.000813412293959037,0.0117930475284729,0.0150909126017621,0.0431812190825411],[4.80866955950915e-28,7.7307596252607e-16,0.000616720399542001,0.00894136102637251,0.0114417666395399,0.0327395330548526],["991/9493/9833/9133/10403/7153/259266/6241/9787/11065/55355/55872/51203/83461/22974/10460/79019/55839/890/983/4085/5080/332/3832/9212/51659/9055/891/4174/9232/1602/2018/4137","991/9133/10403/7153/259266/6241/9787/11065/51203/22974/10460/890/983/332/9212/9055/891/9232/2167/51705","6280/6279/597/7850/6286/366","1307/730/54829/10234/79901/57758/4969","6280/6279/597/7850/6286/366","9493/23397/9787/10460/55839/332/7272/4521/9232"],[33,20,6,7,6,9]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[5,6,7,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="msigdb-analysis" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> MSigDb analysis</h2>
<p><a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a> contains 8 major collections:</p>
<ul>
<li>H: hallmark gene sets</li>
<li>C1: positional gene sets</li>
<li>C2: curated gene sets</li>
<li>C3: motif gene sets</li>
<li>C4: computational gene sets</li>
<li>C5: GO gene sets</li>
<li>C6: oncogenic signatures</li>
<li>C7: immunologic signatures</li>
</ul>
<p>Users can download GMT files from <a href="http://software.broadinstitute.org/gsea/msigdb">Broad Institute</a> and use <code>read.gmt</code> to parse the file to be used in <code>enricher()</code> and <code>GSEA()</code>.</p>
<p>There is an R package, <a href="https://cran.r-project.org/package=msigdbr">msigdbr</a>, that already packed the MSigDB gene sets in tidy data format that can be used directly with <em>clusterProfiler</em>.</p>
<p>It supports several specices:</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="universal-api.html#cb122-1" aria-hidden="true"></a><span class="kw">library</span>(msigdbr)</span>
<span id="cb122-2"><a href="universal-api.html#cb122-2" aria-hidden="true"></a><span class="kw">msigdbr_show_species</span>()</span></code></pre></div>
<pre><code>##  [1] &quot;Bos taurus&quot;               &quot;Caenorhabditis elegans&quot;  
##  [3] &quot;Canis lupus familiaris&quot;   &quot;Danio rerio&quot;             
##  [5] &quot;Drosophila melanogaster&quot;  &quot;Gallus gallus&quot;           
##  [7] &quot;Homo sapiens&quot;             &quot;Mus musculus&quot;            
##  [9] &quot;Rattus norvegicus&quot;        &quot;Saccharomyces cerevisiae&quot;
## [11] &quot;Sus scrofa&quot;</code></pre>
<p>We can retrieve all human gene sets:</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="universal-api.html#cb124-1" aria-hidden="true"></a>m_df &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>)</span>
<span id="cb124-2"><a href="universal-api.html#cb124-2" aria-hidden="true"></a><span class="kw">head</span>(m_df, <span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame</span></code></pre></div>
<pre><code>##    gs_id        gs_name gs_cat      gs_subcat human_gene_symbol species_name
## 1 M12609 AAACCAC_MIR140     C3 MIR:MIR_Legacy             ABCC4 Homo sapiens
## 2 M12609 AAACCAC_MIR140     C3 MIR:MIR_Legacy          ABRAXAS2 Homo sapiens
##   entrez_gene gene_symbol sources
## 1       10257       ABCC4    &lt;NA&gt;
## 2       23172    ABRAXAS2    &lt;NA&gt;</code></pre>
<p>Or specific collection. Here we use C6, oncogenic gene sets as an example:</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="universal-api.html#cb126-1" aria-hidden="true"></a>m_t2g &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;C6&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb126-2"><a href="universal-api.html#cb126-2" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene)</span>
<span id="cb126-3"><a href="universal-api.html#cb126-3" aria-hidden="true"></a><span class="kw">head</span>(m_t2g)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name              entrez_gene
##   &lt;chr&gt;                      &lt;int&gt;
## 1 AKT_UP_MTOR_DN.V1_DN       25864
## 2 AKT_UP_MTOR_DN.V1_DN          95
## 3 AKT_UP_MTOR_DN.V1_DN      137872
## 4 AKT_UP_MTOR_DN.V1_DN         134
## 5 AKT_UP_MTOR_DN.V1_DN       55326
## 6 AKT_UP_MTOR_DN.V1_DN         271</code></pre>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="universal-api.html#cb128-1" aria-hidden="true"></a>em &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE=</span>m_t2g)</span>
<span id="cb128-2"><a href="universal-api.html#cb128-2" aria-hidden="true"></a>em2 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE =</span> m_t2g)</span>
<span id="cb128-3"><a href="universal-api.html#cb128-3" aria-hidden="true"></a><span class="kw">head</span>(em)</span></code></pre></div>
<pre><code>##                                            ID            Description GeneRatio
## RPS14_DN.V1_DN                 RPS14_DN.V1_DN         RPS14_DN.V1_DN    22/183
## GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP    16/183
## VEGF_A_UP.V1_DN               VEGF_A_UP.V1_DN        VEGF_A_UP.V1_DN    15/183
## PRC2_EZH2_UP.V1_DN         PRC2_EZH2_UP.V1_DN     PRC2_EZH2_UP.V1_DN    14/183
## E2F1_UP.V1_UP                   E2F1_UP.V1_UP          E2F1_UP.V1_UP    12/183
## ATF2_S_UP.V1_UP               ATF2_S_UP.V1_UP        ATF2_S_UP.V1_UP    12/183
##                          BgRatio       pvalue     p.adjust       qvalue
## RPS14_DN.V1_DN         184/10897 3.895275e-13 6.583014e-11 5.371379e-11
## GCNP_SHH_UP_LATE.V1_UP 179/10897 5.086172e-08 4.297815e-06 3.506781e-06
## VEGF_A_UP.V1_DN        192/10897 7.792893e-07 4.389997e-05 3.581997e-05
## PRC2_EZH2_UP.V1_DN     189/10897 3.460639e-06 1.462120e-04 1.193010e-04
## E2F1_UP.V1_UP          188/10897 7.640566e-05 2.264800e-03 1.847952e-03
## ATF2_S_UP.V1_UP        189/10897 8.040709e-05 2.264800e-03 1.847952e-03
##                                                                                                                                        geneID
## RPS14_DN.V1_DN         10874/55388/991/9493/1062/4605/9133/23397/79733/9787/55872/83461/54821/51659/9319/9055/10112/4174/5105/2532/7021/79901
## GCNP_SHH_UP_LATE.V1_UP                                      55388/7153/79733/6241/9787/51203/983/9212/1111/9319/9055/3833/6790/4174/3169/1580
## VEGF_A_UP.V1_DN                                                   8318/9493/1062/9133/10403/6241/9787/4085/332/3832/7272/891/23362/2167/10234
## PRC2_EZH2_UP.V1_DN                                               8318/55388/4605/23397/9787/55355/10460/81620/2146/7272/9212/11182/3887/24137
## E2F1_UP.V1_UP                                                                  55388/7153/23397/79733/9787/2146/2842/9212/8208/1111/9055/3833
## ATF2_S_UP.V1_UP                                                               4312/6280/2305/9493/7153/2146/10112/3887/8544/22885/10234/10974
##                        Count
## RPS14_DN.V1_DN            22
## GCNP_SHH_UP_LATE.V1_UP    16
## VEGF_A_UP.V1_DN           15
## PRC2_EZH2_UP.V1_DN        14
## E2F1_UP.V1_UP             12
## ATF2_S_UP.V1_UP           12</code></pre>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="universal-api.html#cb130-1" aria-hidden="true"></a><span class="kw">head</span>(em2)</span></code></pre></div>
<pre><code>##                                  ID       Description setSize enrichmentScore
## CSR_LATE_UP.V1_UP CSR_LATE_UP.V1_UP CSR_LATE_UP.V1_UP     136       0.5952229
## E2F1_UP.V1_UP         E2F1_UP.V1_UP     E2F1_UP.V1_UP     160       0.5311923
## HOXA9_DN.V1_DN       HOXA9_DN.V1_DN    HOXA9_DN.V1_DN     181       0.5003226
## MTOR_UP.V1_UP         MTOR_UP.V1_UP     MTOR_UP.V1_UP     130       0.5740249
## RAF_UP.V1_DN           RAF_UP.V1_DN      RAF_UP.V1_DN     188      -0.5544683
## RPS14_DN.V1_DN       RPS14_DN.V1_DN    RPS14_DN.V1_DN     182       0.5480207
##                         NES pvalue p.adjust      qvalues rank
## CSR_LATE_UP.V1_UP  2.539394  1e-10 3.15e-09 1.684211e-09 2105
## E2F1_UP.V1_UP      2.296655  1e-10 3.15e-09 1.684211e-09 1592
## HOXA9_DN.V1_DN     2.172908  1e-10 3.15e-09 1.684211e-09 1587
## MTOR_UP.V1_UP      2.425658  1e-10 3.15e-09 1.684211e-09 1673
## RAF_UP.V1_DN      -2.276381  1e-10 3.15e-09 1.684211e-09 2035
## RPS14_DN.V1_DN     2.376211  1e-10 3.15e-09 1.684211e-09 1278
##                                     leading_edge
## CSR_LATE_UP.V1_UP tags=52%, list=17%, signal=44%
## E2F1_UP.V1_UP     tags=29%, list=13%, signal=26%
## HOXA9_DN.V1_DN    tags=34%, list=13%, signal=30%
## MTOR_UP.V1_UP     tags=42%, list=13%, signal=37%
## RAF_UP.V1_DN      tags=44%, list=16%, signal=37%
## RPS14_DN.V1_DN    tags=33%, list=10%, signal=30%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                    core_enrichment
## CSR_LATE_UP.V1_UP                                                              55143/2305/4605/6241/11065/55872/983/332/2146/9319/79801/990/11130/1503/51155/81831/2175/2237/56942/6627/84823/64785/6751/55055/51053/3608/3575/22808/8317/4496/7277/1841/5557/4502/4501/4172/5424/10606/11200/4828/6611/9123/10213/93081/7283/4522/8710/29902/4494/865/8662/86/63967/1020/5393/6921/55706/57019/6484/5702/6723/5708/54107/1000/5536/8089/7372/3814/4927/2280/22919
## E2F1_UP.V1_UP                                                                                                                                                                                               55388/7153/23397/79733/9787/2146/2842/9212/8208/1111/9055/3833/4001/419/7083/3149/10733/4173/2237/10926/7298/29078/699/6491/1763/1535/23649/58516/6541/1871/11169/1991/7884/5558/100/92154/23185/4678/9600/10592/7422/9100/1994/29802/142/117178/23076
## HOXA9_DN.V1_DN                                                                                                               8318/55388/79733/55165/9582/9415/9837/81620/55765/5163/990/55215/875/4998/3070/10733/1717/54962/133/4597/9918/29028/84296/1690/1299/6713/7504/64785/55975/10535/55159/6576/55544/6515/55732/11314/586/30001/1749/3008/8530/7545/5558/55076/239/29923/11168/10849/9824/6696/3177/2189/6510/2289/57228/2824/9200/2713/55706/79071/29780
## MTOR_UP.V1_UP                                                                                                                                                         991/6241/11065/983/5163/4174/29968/11274/2001/51155/56649/6352/3070/7037/1717/4175/5214/10926/54962/2643/10437/6713/5111/9982/6614/27346/5268/55247/222255/3948/65260/136/3925/9388/3422/3162/2171/2224/637/467/9466/4939/6513/8807/55287/50814/4522/3434/147/3433/3336/3419/5550/1030/10801
## RAF_UP.V1_DN      23314/221037/1960/51340/55105/10265/22996/9687/5357/1952/12/4602/9231/596/51454/57419/595/4256/23022/8777/7837/7042/3397/57613/323/1831/6451/54843/7358/2353/8773/6938/8991/64699/57007/23389/10560/7227/3485/79068/10769/4254/26018/3480/2674/23327/857/116039/6542/2690/1955/1363/26353/7033/5376/89927/51363/8821/2239/6947/4886/214/3487/3667/5157/54847/54898/7031/6505/57535/10451/18/771/80129/5174/5507/56521/8839/8614/5241/10551/57758
## RPS14_DN.V1_DN                                                                                                                       10874/55388/991/9493/1062/4605/9133/23397/79733/9787/55872/83461/54821/51659/9319/9055/10112/4174/29968/11004/990/5347/55635/9156/875/1894/9700/7083/79094/56992/4998/4288/440/200315/29028/55975/26271/55055/3574/705/23109/51337/6535/51083/4436/6182/586/8877/119/4072/23481/9631/5424/2140/1317/9824/5831/3177/7283/51154</code></pre>
<p>We can test with other collections, for example, using C3 to test whether the genes are up/down-regulated by sharing specific motif.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="universal-api.html#cb132-1" aria-hidden="true"></a>m_t2g &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;C3&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-2"><a href="universal-api.html#cb132-2" aria-hidden="true"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene)</span>
<span id="cb132-3"><a href="universal-api.html#cb132-3" aria-hidden="true"></a><span class="kw">head</span>(m_t2g)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name        entrez_gene
##   &lt;chr&gt;                &lt;int&gt;
## 1 AAACCAC_MIR140       10257
## 2 AAACCAC_MIR140       23172
## 3 AAACCAC_MIR140          81
## 4 AAACCAC_MIR140          90
## 5 AAACCAC_MIR140        8754
## 6 AAACCAC_MIR140       11096</code></pre>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="universal-api.html#cb134-1" aria-hidden="true"></a>em3 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE =</span> m_t2g)</span>
<span id="cb134-2"><a href="universal-api.html#cb134-2" aria-hidden="true"></a><span class="kw">head</span>(em3)</span></code></pre></div>
<pre><code>##                                        ID          Description setSize
## E2F1_Q3                           E2F1_Q3              E2F1_Q3     186
## E2F1_Q6                           E2F1_Q6              E2F1_Q6     183
## HSD17B8_TARGET_GENES HSD17B8_TARGET_GENES HSD17B8_TARGET_GENES     400
## SGCGSSAAA_E2F1DP2_01 SGCGSSAAA_E2F1DP2_01 SGCGSSAAA_E2F1DP2_01     128
## E2F_Q4                             E2F_Q4               E2F_Q4     189
## E2F_Q6                             E2F_Q6               E2F_Q6     186
##                      enrichmentScore      NES       pvalue     p.adjust
## E2F1_Q3                    0.5010091 2.239888 1.000000e-10 1.102000e-07
## E2F1_Q6                    0.4982840 2.220185 1.000000e-10 1.102000e-07
## HSD17B8_TARGET_GENES       0.6375083 3.076806 1.000000e-10 1.102000e-07
## SGCGSSAAA_E2F1DP2_01       0.5448603 2.305239 3.020411e-10 2.496370e-07
## E2F_Q4                     0.4771367 2.136370 4.173413e-10 2.759460e-07
## E2F_Q6                     0.4758875 2.127575 6.401595e-10 3.527279e-07
##                           qvalues rank                   leading_edge
## E2F1_Q3              8.378947e-08 2045 tags=36%, list=16%, signal=31%
## E2F1_Q6              8.378947e-08 2240 tags=39%, list=18%, signal=32%
## HSD17B8_TARGET_GENES 8.378947e-08 1042  tags=36%, list=8%, signal=34%
## SGCGSSAAA_E2F1DP2_01 1.898090e-07 1797 tags=38%, list=14%, signal=33%
## E2F_Q4               2.098128e-07 1663 tags=30%, list=13%, signal=26%
## E2F_Q6               2.681931e-07 1663 tags=30%, list=13%, signal=26%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             core_enrichment
## E2F1_Q3                                                                                                                                                                                                                                                                                                                                                                                                                                                 79733/6241/983/5080/2146/8715/4171/993/990/56992/9768/4998/10733/7037/4175/54962/54954/29028/3015/2643/85236/5111/64785/26271/51053/1869/3925/114/5427/4176/7112/4436/1871/79915/9088/6632/284403/7290/2597/7398/4172/5424/124222/1633/10849/9824/4678/6646/29902/63967/56886/6059/23234/1786/204/51087/23636/2048/83463/10393/79173/688/3007/79677/9462/56946/9221
## E2F1_Q6                                                                                                                                                                                                                                                                                                                                                                                                                                79733/6241/983/5080/81620/2146/4171/993/990/3159/9768/4998/4175/4173/29028/3015/4609/85236/5111/64785/26271/51053/1869/3925/5427/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7374/9824/2189/4678/2289/6646/29902/50/63967/56886/23234/6839/1786/204/51087/23636/11335/1174/83463/10492/79173/688/79677/9462/56946/9221/5933/26135/988/5591
## HSD17B8_TARGET_GENES 55143/55388/991/2305/9493/1062/4605/9833/9133/10403/7153/23397/79733/259266/6241/55165/9787/11065/55355/9582/220134/55872/83461/10460/4751/10635/55839/890/9415/983/54821/4085/9837/5080/332/7272/64151/2842/9212/51659/9319/9055/3833/146909/10112/51514/4174/9232/1033/9928/3161/11004/993/4603/79801/990/5347/55215/701/55723/51512/55635/9156/11130/10024/57405/10615/1894/2491/8438/9700/5888/7083/898/3149/56992/9768/4288/10733/1163/4175/5307/2237/29899/55010/59336/56942/5984/29028/3838/1058/54478/3015/699/6491/81611/1063/27346/64785/9401/26271/51026/641/1869/10535/1029/28998/1763/8970/54892/55159/864/93594/1789/4176/7111/3148/116832/9585/9735/55732/81624/23310/1871/1031/79915/995/10051/30012/1104/80178/78995/7468/284403/5230/8549/7884/5558/4172/5424/79866/3182/56902/83990
## SGCGSSAAA_E2F1DP2_01                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     79733/6241/4171/993/990/3159/9768/4175/4173/29028/3015/4609/85236/5111/64785/26271/51053/1869/5427/4176/5902/4436/1871/79915/9088/6632/284403/7290/4172/5424/124222/1633/7374/2189/4678/2289/50/63967/56886/23234/6839/1786/204/51087/23636/1174/83463/10492/79173
## E2F_Q4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 8318/79733/6241/983/5080/81620/2146/4171/993/990/5888/9768/4175/4173/29028/3015/85236/5111/64785/26271/51053/1869/3925/5427/23649/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7473/7374/9824/7353/4678/6646/50/63967/56886/23234/1786/204/6427/51087/23636
## E2F_Q6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      8318/79733/6241/983/5080/81620/2146/4171/993/990/5888/9768/4175/4173/29028/3015/85236/5111/64785/26271/51053/1869/3925/5427/23649/4176/7112/5902/4436/1871/79915/9088/6632/284403/7290/2597/4172/5424/124222/1633/10849/7374/9824/7353/4678/6646/50/63967/56886/23234/1786/204/6427/51087/23636</code></pre>
<!--


# DAVID functional analysis

[clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) provides enrichment and GSEA analysis with GO, KEGG, DO and Reactome pathway supported internally, some user may prefer GO and KEGG analysis with DAVID[@huang_david_2007] and still attracted by the visualization methods provided by [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler)[@paranjpe_genome_wid_2013]. To bridge the gap between DAVID and clusterProfiler, we implemented `enrichDAVID`. This function query enrichment analysis result from DAVID webserver via [RDAVIDWebService](https://www.bioconductor.org/packages/RDAVIDWebService)[@fresno_rdavidwebservice_2013] and stored the result as an `enrichResult` instance, so that we can use all the visualization functions in [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) to visualize DAVID results. `enrichDAVID` is fully compatible with `compareCluster` function and comparing enrichment results from different gene clusters is now available with DAVID.


```r
david <- enrichDAVID(gene = gene,
                     idType = "ENTREZ_GENE_ID",
                     listType = "Gene",
                     annotation = "KEGG_PATHWAY",
                     david.user = "clusterProfiler@hku.hk")
```

DAVID Web Service has the following limitations:

+ A job with more than 3000 genes to generate gene or term cluster report will not be handled by DAVID due to resource limit.
+ No more than 200 jobs in a day from one user or computer.
+ DAVID Team reserves right to suspend any improper uses of the web service without notice.

For more details, please refer to [http://david.abcc.ncifcrf.gov/content.jsp?file=WS.html](http://david.abcc.ncifcrf.gov/content.jsp?file=WS.html).

As user has limited usage, please [register](http://david.abcc.ncifcrf.gov/webservice/register.htm) and use your own user account to run `enrichDAVID`.


-->
</div>
<div id="msigdb" class="section level2" number="12.4">
<h2><span class="header-section-number">12.4</span> MSigDb</h2>
<p>The MSigDB is a collection of annotated gene sets, it include 8 major collections:</p>
<ul>
<li>H: hallmark gene sets</li>
<li>C1: positional gene sets</li>
<li>C2: curated gene sets</li>
<li>C3: motif gene sets</li>
<li>C4: computational gene sets</li>
<li>C5: GO gene sets</li>
<li>C6: oncogenic signatures</li>
<li>C7: immunologic signatures</li>
</ul>
<p>Users can use <code>enricher</code> and <code>GSEA</code> function to analyze gene set collections downloaded from Molecular Signatures Database (<a href="http://www.broadinstitute.org/gsea/msigdb/index.jsp">MSigDb</a>). <a href="https://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a> provides a function, <code>read.gmt</code>, to parse the <a href="www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29">gmt file</a> into a <em>TERM2GENE</em> <code>data.frame</code> that is ready for both <code>enricher</code> and <code>GSEA</code> functions.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="universal-api.html#cb136-1" aria-hidden="true"></a><span class="kw">data</span>(geneList, <span class="dt">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb136-2"><a href="universal-api.html#cb136-2" aria-hidden="true"></a>gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]</span>
<span id="cb136-3"><a href="universal-api.html#cb136-3" aria-hidden="true"></a></span>
<span id="cb136-4"><a href="universal-api.html#cb136-4" aria-hidden="true"></a>gmtfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;c5.cc.v5.0.entrez.gmt&quot;</span>, <span class="dt">package=</span><span class="st">&quot;clusterProfiler&quot;</span>)</span>
<span id="cb136-5"><a href="universal-api.html#cb136-5" aria-hidden="true"></a>c5 &lt;-<span class="st"> </span><span class="kw">read.gmt</span>(gmtfile)</span>
<span id="cb136-6"><a href="universal-api.html#cb136-6" aria-hidden="true"></a></span>
<span id="cb136-7"><a href="universal-api.html#cb136-7" aria-hidden="true"></a>egmt &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE=</span>c5)</span>
<span id="cb136-8"><a href="universal-api.html#cb136-8" aria-hidden="true"></a><span class="kw">head</span>(egmt)</span></code></pre></div>
<pre><code>##                                                ID              Description
## SPINDLE                                   SPINDLE                  SPINDLE
## MICROTUBULE_CYTOSKELETON MICROTUBULE_CYTOSKELETON MICROTUBULE_CYTOSKELETON
## CYTOSKELETAL_PART               CYTOSKELETAL_PART        CYTOSKELETAL_PART
## SPINDLE_MICROTUBULE           SPINDLE_MICROTUBULE      SPINDLE_MICROTUBULE
## MICROTUBULE                           MICROTUBULE              MICROTUBULE
## CYTOSKELETON                         CYTOSKELETON             CYTOSKELETON
##                          GeneRatio  BgRatio       pvalue     p.adjust
## SPINDLE                      11/82  39/5270 7.667674e-12 5.214018e-10
## MICROTUBULE_CYTOSKELETON     16/82 152/5270 8.449298e-10 2.872761e-08
## CYTOSKELETAL_PART            15/82 235/5270 2.414879e-06 5.237096e-05
## SPINDLE_MICROTUBULE           5/82  16/5270 3.080645e-06 5.237096e-05
## MICROTUBULE                   6/82  32/5270 7.740446e-06 1.052701e-04
## CYTOSKELETON                 16/82 367/5270 1.308357e-04 1.482805e-03
##                                qvalue
## SPINDLE                  4.197043e-10
## MICROTUBULE_CYTOSKELETON 2.312439e-08
## CYTOSKELETAL_PART        4.215619e-05
## SPINDLE_MICROTUBULE      4.215619e-05
## MICROTUBULE              8.473751e-05
## CYTOSKELETON             1.193589e-03
##                                                                                                  geneID
## SPINDLE                                           991/9493/9787/22974/983/332/3832/7272/9055/6790/24137
## MICROTUBULE_CYTOSKELETON 991/9493/9133/7153/9787/22974/4751/983/332/3832/7272/9055/6790/24137/4137/7802
## CYTOSKELETAL_PART             991/9493/7153/9787/22974/4751/983/332/3832/7272/9055/6790/24137/4137/7802
## SPINDLE_MICROTUBULE                                                             983/332/3832/9055/24137
## MICROTUBULE                                                                983/332/3832/9055/24137/4137
## CYTOSKELETON             991/9493/9133/7153/9787/22974/4751/983/332/3832/7272/9055/6790/24137/4137/7802
##                          Count
## SPINDLE                     11
## MICROTUBULE_CYTOSKELETON    16
## CYTOSKELETAL_PART           15
## SPINDLE_MICROTUBULE          5
## MICROTUBULE                  6
## CYTOSKELETON                16</code></pre>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="universal-api.html#cb138-1" aria-hidden="true"></a>egmt2 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE=</span>c5, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</span>
<span id="cb138-2"><a href="universal-api.html#cb138-2" aria-hidden="true"></a><span class="kw">head</span>(egmt2)</span></code></pre></div>
<pre><code>##                                                                    ID
## CHROMOSOMAL_PART                                     CHROMOSOMAL_PART
## CHROMOSOME                                                 CHROMOSOME
## PROTEINACEOUS_EXTRACELLULAR_MATRIX PROTEINACEOUS_EXTRACELLULAR_MATRIX
## EXTRACELLULAR_MATRIX                             EXTRACELLULAR_MATRIX
## SPINDLE                                                       SPINDLE
## CHROMOSOMEPERICENTRIC_REGION             CHROMOSOMEPERICENTRIC_REGION
##                                                           Description setSize
## CHROMOSOMAL_PART                                     CHROMOSOMAL_PART      87
## CHROMOSOME                                                 CHROMOSOME     112
## PROTEINACEOUS_EXTRACELLULAR_MATRIX PROTEINACEOUS_EXTRACELLULAR_MATRIX      93
## EXTRACELLULAR_MATRIX                             EXTRACELLULAR_MATRIX      95
## SPINDLE                                                       SPINDLE      37
## CHROMOSOMEPERICENTRIC_REGION             CHROMOSOMEPERICENTRIC_REGION      30
##                                    enrichmentScore       NES       pvalue
## CHROMOSOMAL_PART                         0.6306975  2.493293 1.000000e-10
## CHROMOSOME                               0.6136947  2.504790 1.000000e-10
## PROTEINACEOUS_EXTRACELLULAR_MATRIX      -0.6355317 -2.358895 1.000000e-10
## EXTRACELLULAR_MATRIX                    -0.6229461 -2.315436 1.289663e-10
## SPINDLE                                  0.7512846  2.520608 1.513292e-09
## CHROMOSOMEPERICENTRIC_REGION             0.7846974  2.510922 7.146580e-09
##                                        p.adjust      qvalues rank
## CHROMOSOMAL_PART                   6.254867e-09 4.887145e-09 1681
## CHROMOSOME                         6.254867e-09 4.887145e-09 1905
## PROTEINACEOUS_EXTRACELLULAR_MATRIX 6.254867e-09 4.887145e-09 1473
## EXTRACELLULAR_MATRIX               6.254867e-09 4.887145e-09 1473
## SPINDLE                            5.871574e-08 4.587665e-08  759
## CHROMOSOMEPERICENTRIC_REGION       2.077266e-07 1.623040e-07  730
##                                                      leading_edge
## CHROMOSOMAL_PART                   tags=38%, list=13%, signal=33%
## CHROMOSOME                         tags=38%, list=15%, signal=33%
## PROTEINACEOUS_EXTRACELLULAR_MATRIX tags=49%, list=12%, signal=44%
## EXTRACELLULAR_MATRIX               tags=48%, list=12%, signal=43%
## SPINDLE                             tags=43%, list=6%, signal=41%
## CHROMOSOMEPERICENTRIC_REGION        tags=40%, list=6%, signal=38%
##                                                                                                                                                                                                                                                                core_enrichment
## CHROMOSOMAL_PART                                                                                   1062/10403/4085/81620/332/4171/701/11130/11339/3070/54962/5984/1058/699/1063/5111/55055/55506/4176/7112/2296/79980/5982/10051/1104/4172/2072/5983/10592/8914/1788/4613/3619
## CHROMOSOME                                       1062/10403/7153/4085/81620/332/1111/4171/701/11130/5888/11339/3070/54962/5984/9918/1058/699/1063/5111/55055/55506/4176/7112/2296/3148/79980/5982/10051/1104/4172/2072/7283/5983/10592/8914/1788/6839/4613/3619/5901/7273/6119
## PROTEINACEOUS_EXTRACELLULAR_MATRIX 80781/1490/1306/8425/8082/4256/8910/3912/1278/4060/283/30008/5549/22795/10516/1293/1295/2192/1281/50509/4319/1290/11096/2202/2199/6444/1294/1462/1289/3908/3909/4053/8292/1842/10631/2331/3730/3913/1300/2200/1287/7060/1307/1311/4148/1101
## EXTRACELLULAR_MATRIX               80781/1490/1306/8425/8082/4256/8910/3912/1278/4060/283/30008/5549/22795/10516/1293/1295/2192/1281/50509/4319/1290/11096/2202/2199/6444/1294/1462/1289/3908/3909/4053/8292/1842/10631/2331/3730/3913/1300/2200/1287/7060/1307/1311/4148/1101
## SPINDLE                                                                                                                                                                                        991/9493/9787/22974/983/332/3832/7272/9055/6790/24137/699/1063/26271/29901/9735
## CHROMOSOMEPERICENTRIC_REGION                                                                                                                                                                                      1062/10403/4085/332/701/11130/3070/1058/699/1063/55055/79980</code></pre>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-yu2012">
<p>Yu, Guangchuang, Li-Gen Wang, Yanyan Han, and Qing-Yu He. 2012. ‚ÄúclusterProfiler: an R Package for Comparing Biological Themes Among Gene Clusters.‚Äù <em>OMICS: A Journal of Integrative Biology</em> 16 (5): 284‚Äì87. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="meshes-enrichment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chipseeker-enrichment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
